---
title: "BMvsFL_HSC_scATAC_analyses_downscaled_bams"
output: html_document
---
``



```{r cellranger_counts, on Lsens2}
#Use the, by BjÃ¶rn, demultiplexed fastqs and setup one SLURM-script for each.
#Set up the script and run for samples ABM_CD41neg ABM_CD41pos ABM_MkP and FL_HSC:

cell_ranger_ref="/projects/fs5/jonun/scatacseq/10Xindices/refdata-cellranger-atac-mm10-1.2.0"
fastq_path="/projects/fs5/jonun/scatacseq/191204_A00681_0057_BHHTF7DRXX_jy_HSC/cellranger_out/outs/fastq_path/HHTF7DRXX/ABM_CD41neg"
sample_name="ABM_CD41neg"
# Load the modules
module purge
module load cellranger-atac/1.0.1

cellranger-atac count --id=${sample_name} --reference=${cell_ranger_ref} --fastqs=${fastq_path} --sample=${sample_name}


sbatch 191209_cellranger_count_scatac_ABM_CD41neg.skript

```

## Including Plots

You can also embed plots, for example:

```{r preprocess_snap_atac, echo=FALSE}
#Catenate R1,R3 and R2 as above
#Create a preprocess script and run as follows
bash snapatac_preprocess.skript  2>&1 | tee 190930_snapatac_preprocessins.log
#Do not forget to change the date of the log to 191210

```
```{r snap_atac_barcode_filtering, echo=FALSE}
library(SnapATAC)
library(viridisLite)
library(ggplot2)
library(stringr)
#Create an experiment from multiple samples.
file.list = c("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/FL_HSC/FL_HSC.snap/FL_HSC.snap","/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/ABM_CD41neg/ABM_CD41neg.snap/ABM_CD41neg.sub.snap","/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/ABM_CD41pos/ABM_CD41pos.snap/ABM_CD41pos.sub.snap","/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/ABM_MkP/ABM_MkP.snap/ABM_MkP.sub.snap")

sample.list=c("FL_HSC","CD41neg","CD41pos","MkP")

x.sp = createSnap(file=file.list, sample=sample.list)


#Instead of the 10x barcode filtering below, try SnapATACs "internal" filtering to see how many cells we get.

library(GenomicRanges);
library(ggplot2)
x.sp = addBmatToSnap(x.sp, bin.size=5000,do.par = TRUE, num.cores=12)

#mm10 promoter file from scATAC-pro is used for the filtering step: https://github.com/wbaopaul/scATAC-pro/tree/master/annotation
promoter.df = read.table("/mnt/data/common/jonas/scatac/scATAC-pro_1.1.1/annotation/mm10_promoter.bed");
promoter.gr = GRanges(promoter.df[,1], IRanges(promoter.df[,2], promoter.df[,3]));
ov = findOverlaps(x.sp@feature, promoter.gr);
idy = queryHits(ov);
cov = log10(SnapATAC::rowSums(x.sp, mat="bmat")+1)
promoter_ratio = SnapATAC::rowSums(x.sp[,idy, mat="bmat"], mat="bmat") / SnapATAC::rowSums(x.sp, mat="bmat");
#Plot the results
barcodes_to_plot <- cbind(as.data.frame(cov), as.data.frame(promoter_ratio))
colnames(barcodes_to_plot)<-c("cov", "promoter_ratio")
p1 = ggplot(data =barcodes_to_plot, aes(x= cov, y= promoter_ratio)) + geom_point(size=0.1, col="grey") + ggtitle("10X HSCs") + ylim(0, 1) + xlim(0, 6) + labs(x = "log10(UMI)", y="promoter ratio") 

p1 + geom_hline(yintercept = 0.27) + geom_vline(xintercept = 4.15)

idx = which(promoter_ratio > 0.27 & promoter_ratio < 0.8 & cov > 4.15);
x.sp2 <-x.sp
x.sp2 = x.sp2[idx,];

#Make the matrix binary:
#Make matrix binary
x.sp2 = makeBinary(x.sp2, mat="bmat")


```

```{r black_list_and_bin_coverage_filtering, echo=FALSE}
#Bin filtering
#Remove blacklist regions
system("wget http://mitra.stanford.edu/kundaje/akundaje/release/blacklists/mm10-mouse/mm10.blacklist.bed.gz")
library(GenomicRanges)
black_list = read.table("mm10.blacklist.bed.gz")
black_list.gr = GRanges(black_list[,1],  IRanges(black_list[,2], black_list[,3]))
idy = queryHits(findOverlaps(x.sp2@feature, black_list.gr))
if(length(idy) > 0){x.sp2 = x.sp2[,-idy, mat="bmat"]}

#Filter for bin coverage
bin.cov = log10(Matrix::colSums(x.sp2@bmat)+1)
hist(bin.cov[bin.cov > 0], xlab="log10(bin cov)", main="log10(Bin Cov)", col="lightblue",  xlim=c(0, 5))
bin.cutoff = quantile(bin.cov[bin.cov > 0], 0.95);
idy = which(bin.cov <= bin.cutoff & bin.cov > 0);
x.sp2 = x.sp2[, idy, mat="bmat"];

```
```{r snap_atac_clustering, echo=FALSE}
# Dimensionality reduction

x.sp2 = runDiffusionMaps(obj=x.sp2, input.mat="bmat", num.eigs=50)

plotDimReductPW(obj=x.sp2, eigs.dims=1:50,point.size=0.3, point.color="grey", point.shape=19,point.alpha=0.6, down.sample=5000, pdf.file.name=NULL,  pdf.height=7, pdf.width=7)
# A little tricky but choose the 10 first dimensions since that is a little blob shaped after.

x.sp3 <- x.sp2
#Graph based clusterning
x.sp3 = makeBinary(x.sp3, mat="bmat")
x.sp3 = runKNN(obj=x.sp3,eigs.dims=1:10,k=50) #We set it to 50 to make a more stringent clustering. More similar neigbours,
x.sp3=runCluster(obj=x.sp3,tmp.folder=tempdir(),louvain.lib="R-igraph",seed.use=10);
x.sp3@metaData$cluster = x.sp3@cluster
  
#Visualization
x.sp3 = runViz(obj=x.sp3, tmp.folder=tempdir(),dims=2, eigs.dims=1:10, method="umap",seed.use=10)

par(mfrow = c(2, 2))

#Some cluster and qc-plotting.
plotViz(obj=x.sp3,method="umap",  main="HSCvsMkP", point.color=x.sp3@sample,  point.size=0.2, point.shape=19, point.alpha=0.8, text.add=TRUE,text.size=1.5, text.color="black", text.halo.add=TRUE, text.halo.color="white",text.halo.width=0.2,legend.add=FALSE)
  
plotViz(obj=x.sp3,method="umap",  main="HSCvsMkP", point.color=x.sp3@cluster,  point.size=0.2, point.shape=19, point.alpha=0.8, text.add=TRUE,text.size=1.5, text.color="black", text.halo.add=TRUE, text.halo.color="white",text.halo.width=0.2,legend.add=FALSE)

plotFeatureSingle( obj=x.sp3,feature.value=log(x.sp3@metaData$UQ+1,10), method="umap", main="10X HSC Read Depth Downscaled",point.size=0.1, point.shape=19, quantiles=c(0.2, 0.8))

plotFeatureSingle(obj=x.sp3.sample_peaks, feature.value=Matrix::rowSums(x.sp3.sample_peaks@pmat) / x.sp3.sample_peaks@metaData$UQ,method="umap",  main="HSCvsMkP FRiP", point.size=0.1,  point.shape=19, quantiles=c(0.2, 0.8))
```

```{r mm10_annotation_and_gene_access_plotting, echo=FALSE}
#Gene based annotation and plotting:

system("wget http://renlab.sdsc.edu/r3fang/share/github/Mouse_Brain_10X/gencode.vM16.gene.bed");
genes = read.table("../gencode.vM16.gene.bed");
genes.gr = GRanges(genes[,1],  IRanges(genes[,2], genes[,3]), name=genes[,4]
  );
marker.genes = c("Lin28b", "Gata1", "Spi1","Nfe2", "Cebpa","Itga2b", "Vwf", "Gp9", "Itgb3" )
genes.sel.gr <- genes.gr[which(genes.gr$name %in% marker.genes)]
# re-add the cell-by-bin matrix to the snap object;
x.sp3 = addBmatToSnap(x.sp3);
promoter_ratio = SnapATAC::rowSums(x.sp3[,idy, mat="bmat"], mat="bmat") / SnapATAC::rowSums(x.sp3, mat="bmat");
x.sp3@metaData$promoter_ratio <- promoter_ratio
x.sp3 = makeBinary(x.sp3, mat="bmat")
x.sp3 = createGmatFromMat(obj=x.sp3,  input.mat="bmat",genes=genes.sel.gr, do.par=TRUE, num.cores=24)
# normalize the cell-by-gene matrix
#x.sp3.sample_peaks = scaleCountMatrix(obj=x.sp3.sample_peaks, cov=x.sp3.sample_peaks@metaData$passed_filters + 1,mat="gmat",method = "RPM")
x.sp3 = scaleCountMatrix(obj=x.sp3, cov=x.sp3@metaData$promoter_ratio + 1,mat="gmat",method = "RPM")

# smooth the cell-by-gene matrix
x.sp3 = runMagic( obj=x.sp3,input.mat="gmat",step.size=3)
par(mfrow = c(3, 3))

#Plot the marker genes above.    
for(i in 1:9){
  
    plotFeatureSingle( obj=x.sp3,feature.value=x.sp3@gmat[, marker.genes[i]], method="umap",  main=marker.genes[i], point.size=0.1,  point.shape=19, quantiles=c(0, 1) )
  
  }   
 
```



```{r hierchial_clustering, echo=FALSE}
#Pool 
# calculate the ensemble signals for each cluster
ensemble.x.sp3 = lapply(split(seq(length(x.sp3@cluster)), x.sp3@cluster), function(x){SnapATAC::colMeans(x.sp3[x,], mat="bmat");})

#cluster using 1-cor as distance  
hc.x.sp3 = hclust(as.dist(1 - cor(t(do.call(rbind, ensemble.x.sp3)))), method="ward.D2")

plotViz(
    obj=x.sp3,
    method="umap", 
    main="HSC vs MkP Cluster",
    point.color=x.sp3@cluster, 
    point.size=0.2, 
    point.shape=19, 
    point.alpha=0.8, 
    text.add=TRUE,
    text.size=1.5,
    text.color="black",
    text.halo.add=TRUE,
    text.halo.color="white",
    text.halo.width=0.2,
    down.sample=10000,
    legend.add=FALSE
    )

plot(hc.x.sp3, hang=-1, xlab="")
```

```{r peak_calling_clusters, echo=FALSE}
system("conda activate base")
system("which snaptools")
#/opt/anaconda3/bin/snaptools
system("which macs2")
# /opt/anaconda3/bin/macs2

#Iterate over all the clusters and call peaks.
clusters.sel.peaks = names(table(x.sp3@cluster))[which(table(x.sp3@cluster) > 200)];
peaks.ls = mclapply(seq(clusters.sel.peaks), function(i){
    print(clusters.sel.peaks[i]);
    runMACS(
        obj=x.sp3[which(x.sp3@cluster==clusters.sel.peaks[i]),], 
        output.prefix=paste0("HSCvsMkP_peaks_cluster.", gsub(" ", "_", clusters.sel.peaks)[i]),
        path.to.snaptools="/opt/anaconda3/bin/snaptools",
        path.to.macs="/opt/anaconda3/bin/macs2",
        gsize="mm", 
        buffer.size=500, 
        num.cores=4,
        macs.options="--nomodel --shift 100 --ext 200 --qval 5e-2 -B --SPMR",
        tmp.folder=tempdir()
   );
 }, mc.cores=8);




```

```{r peak_calling_sample_based, echo=FALSE}
system("conda activate base")
system("which snaptools")
#/opt/anaconda3/bin/snaptools
system("which macs2")
# /opt/anaconda3/bin/macs2
setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled")
#Iterate over all the samples and call peaks.
samples.sel.peaks = names(table(x.sp3@sample))[which(table(x.sp3@sample) > 200)];
peaks.sample = mclapply(seq(samples.sel.peaks), function(i){
    print(samples.sel.peaks[i]);
    runMACS(
        obj=x.sp3[which(x.sp3@sample==samples.sel.peaks[i]),], 
        output.prefix=paste0("HSCvsMkP_peaks_samples.", gsub(" ", "_", samples.sel.peaks)[i]),
        path.to.snaptools="/opt/anaconda3/bin/snaptools",
        path.to.macs="/opt/anaconda3/bin/macs2",
        gsize="mm", 
        buffer.size=500, 
        num.cores=12,
        macs.options="--nomodel --shift 100 --ext 200 --qval 5e-2 -B --SPMR",
        tmp.folder=tempdir()
   );
 }, mc.cores=4);

#The chromosome names are weird in these peak files so before peak calling and importing them back in I remove all strange characters in Excel-

#Work with sample and not cluster peaks for now.
# assuming all .narrowPeak files in the current folder are generated from the clusters
setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/peaks_sample_based_downscaled/")
peaks.names = system("ls | grep narrowPeak", intern=TRUE)
peak.gr.ls = lapply(peaks.names, function(x){
    peak.df = read.table(x)
    GRanges(peak.df[,1], IRanges(peak.df[,2], peak.df[,3]))
  })

peak.gr = reduce(Reduce(c, peak.gr.ls));
peak.gr

peaks.df <- as.data.frame(peak.gr)[,1:3]


#Create a cell-by-peak matrix
library(GenomicRanges) #Step below is dependent on Genomic Ranges.
x.sp3.sample_peaks = createPmat(x.sp3.sample_peaks, peak.gr,  do.par = TRUE, num.cores = 4)

plotFeatureSingle(obj=x.sp3.sample_peaks, feature.value=Matrix::rowSums(x.sp3.sample_peaks@pmat) / x.sp3.sample_peaks@metaData$UQ,method="umap",  main="HSCvsMkP FRiP", point.size=0.1,  point.shape=19, quantiles=c(0.01, 0.99))

x.sp3.sample_peaks = makeBinary(x.sp3.sample_peaks, mat="pmat")


save.image("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/BMvsFL_HSC_snapatac_analyses_downscaled_bams.RData")
```

```{r pseudo_time_analysis, echo=FALSE}

#Export as a Seurat object. Use same eigs dims as above. This does not work well because very little data is transfered to the seurat object. Can I create a single cell experiment-oject from the pmat and start there? 
#Convert pmat to a matrix.

peak.matrix <- as(object = x.sp3.sample_peaks@pmat, Class = 'matrix')
peak.dataframe <- as.data.frame(t(peak.matrix))

#assign column names corresponding to the cell barcodes.
colnames(peak.dataframe) <- x.sp3.sample_peaks@sample

#Assign rownames (they'll be lost), and a new column as a factor of the peaks in Cicero format.
rownames(peak.dataframe) <- paste0(peaks.df$seqnames,"_",peaks.df$start,"_",peaks.df$end)
head(peak.dataframe)

#Create a singlecellexperiment
library(SingleCellExperiment)
library(scater)
sce <- SingleCellExperiment(assays = list(counts = as.matrix(peak.dataframe), logcounts = as.matrix(peak.dataframe)))
sce$cell_type <- colnames(sce)
sce$snap_cluster <- x.sp3.sample_peaks@cluster

########################################################################################
############### Plot how variability changes over the different peaks ##################
########################################################################################

peak.var <- apply(peak.dataframe,1,var)
peak.var <- as.data.frame(peak.var)
peak.var  <- peak.var[order(peak.var$peak.var, decreasing = T),]
peak.var <- as.data.frame(peak.var)
peak.var$counter <- 1:198127

plot(peak.var$counter,peak.var$peak.var, type="l", col="black", lwd=2, xlab="Peak", ylab="Variance")
abline(v=c(10000,20000), col=c("blue", "red"), lty=c(1,2), lwd=c(1, 3))
########################################################################################
########################################################################################
########################################################################################

#Run PCA with scater package.
library(scater)
sce <- runPCA(sce, ncomponents = 20, ntop = 10000)
sce <- runDiffusionMap(sce, ncomponents = 20, ntop = 10000)


#Should I rerun this with 10000 peaks to match the 9000 in the NFE2 Gata1 peak list? The answer was yes.

#Test to perform a UMAP reduction on the PCA from sce-
sce2 <- sce
sce2 <- SingleCellExperiment(assays = list(counts = t(as.matrix(sce@reducedDims$PCA)), logcounts = t(as.matrix(sce@reducedDims$PCA))))
sce2<- runUMAP(sce2 )
sce2 <- runDiffusionMap(sce2)
sce2$cell_type <- x.sp3.sample_peaks@sample
sce2$snap_cluster <- x.sp3.sample_peaks@cluster
plotReducedDim(sce2, use_dimred = "UMAP", colour_by = "cell_type")

#Plot the UMAP colored by cell type-
plotReducedDim(sce, use_dimred = "PCA", colour_by = "cell_type")

#Apply slingshot
library(slingshot)
#Both DiffusionMaps and PCA performs poorly.
sce <- slingshot(sce, reducedDim = "UMAP", clusterLabels = c(rep("FL_HSC",2403), rep("CD41neg",2310), rep("CD41pos",1123), rep("MkP",2986)))
sce <- slingshot(sce, reducedDim = "DiffusionMap", clusterLabels = c(rep("FL_HSC",2403), rep("CD41neg",2310), rep("CD41pos",1123), rep("MkP",2986)))

sce2 <- slingshot(sce2, reducedDim = "UMAP", start.clus = "FL_HSC",end.clus = "MkP", clusterLabels = c(rep("FL_HSC",2403), rep("CD41neg",2310), rep("CD41pos",1123), rep("MkP",2986) ) )
library(ggplot2)
library(ggbeeswarm)
library(ggthemes)

#Plot pseudotime vs cell-type and cluster.

ggplot(as.data.frame(colData(sce2)), aes(x = sce2$slingPseudotime_1, y = factor( cell_type, levels=c('FL_HSC', 'CD41neg', 'CD41pos', 'MkP') ), 
                              colour = cell_type)) +
    geom_quasirandom(groupOnX = FALSE) +
    theme_classic() +
    xlab("Slingshot pseudotime") + ylab("Timepoint") +
    ggtitle("Cells ordered by Slingshot pseudotime var_top10000_peaks")

ggsave(filename = "slingshot_pseudotime_order_cell_type_var_top10000_peaks_downscaled_enforced_start_end_sample.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 20, height = 15)

ggplot(as.data.frame(colData(sce2)), aes(x = sce2$slingPseudotime_1, y = snap_cluster, 
                              colour = snap_cluster)) +
    geom_quasirandom(groupOnX = FALSE) +
    theme_classic() +
    xlab("Slingshot pseudotime") + ylab("Timepoint") +
    ggtitle("Cells ordered by Slingshot pseudotime var_top10000_peaks")

ggsave(filename = "slingshot_pseudotime_order_snap_cluster_var_top10000_peaks.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/Snap_Figures/", units = c("cm"), width = 20, height = 15)

colors <- colorRampPalette(rev(brewer.pal(11,'Spectral')[-6]))(100)
plotcol <- colors[cut(sce2$slingPseudotime_1, breaks=100)]
plot(reducedDims(sce2)$UMAP, col = plotcol, pch=16, asp = 1)
lines(SlingshotDataSet(sce2), lwd=2, col='black')

#I may want to revisit the pseudotime analysis with a more refined peak list later.
colors <- colorRampPalette(rev(brewer.pal(11,'Spectral')[-6]))(100)
plotcol <- colors[cut(sce2$slingPseudotime_1, breaks=100)]

plot(reducedDims(sce2)$UMAP, col = "black", bg=plotcol, pch=21, cex=0.6, xlim=c(-5,10), lwd=0.5, axes=FALSE)
axis(side = 1, at = c(-5,-2.5,0,2.5,5,7.5,10))
axis(side = 2, at = c(seq(-6,10, by = 2)))
lines(SlingshotDataSet(sce2), lwd=2, col='black')



cell_type_color <- c("blue", "green", "red", "goldenrod1")
plot(reducedDims(sce2)$UMAP, col = "black", bg = cell_type_color[factor(sce2$cell_type)], pch=21, cex=c(rep(0.6,2403), rep(0.6,2310), rep(0.4,1123), rep(0.6,2986)), lwd=0.5, xlim=c(-5,10), axes=FALSE)
axis(side = 1, at = c(-5,-2.5,0,2.5,5, 7.5,10))
axis(side = 2, at = c(seq(-6,10, by = 2)))



```

```{r plot_clusters_in_cell_types, echo=FALSE}
#Calculate clusters per cell type

table(x.sp3.sample_peaks@cluster)

sample.in.cluster <- cbind( as.data.frame( table(x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="FL_HSC")])), as.data.frame( table(x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="CD41neg")])), as.data.frame( table(x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="CD41pos")])) , as.data.frame(( table(x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="MkP")]))))
sample.in.cluster <- sample.in.cluster[,c(2,4,6,8)]

colnames(sample.in.cluster) <- c("FL_HSC","CD41neg", "CD41pos", "MkP")
rownames(sample.in.cluster) <- c("cluster1","cluster2","cluster3","cluster4","cluster5","cluster6","cluster7","cluster8","cluster9","cluster10","cluster11")
sample.in.cluster$cluster <- rownames(sample.in.cluster)


library(ggplot2)
library(reshape2)

sample.in.cluster <- melt(sample.in.cluster, id.vars = "cluster")
sample.in.cluster$cluster <- factor(sample.in.cluster$cluster, levels = c("cluster1","cluster2","cluster3","cluster4","cluster5","cluster6","cluster7","cluster8","cluster9","cluster10","cluster11"))

ggplot(sample.in.cluster, aes(variable,value)) + geom_bar(position="fill", stat="identity", aes(fill = cluster)) + theme_bw()
ggsave(filename = "snap_atac_clusters_per_sample.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/Figures", units = c("cm"), width = 20, height = 15)

ggplot(sample.in.cluster, aes(cluster,value)) + geom_bar(position="fill", stat="identity", aes(fill = variable)) + theme_bw() +  theme(axis.text.x=element_text(angle=45,hjust=1))
ggsave(filename = "snap_atac_samples_per_cluster.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/Figures", units = c("cm"), width = 20, height = 15)

```

```{r DAR_identification, echo=FALSE}

#Base the differential peak calling on samples and not clusters so first I have to transfer each cluster to an integer.
#First save the cluster in a separate variable
x.sp3.cluster <- x.sp3.sample_peaks@cluster

#DARs need to be called on integers in the cluster object so over write it with sample
x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="FL_HSC")] <- 1
x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="CD41pos")] <- 2
x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="CD41neg")] <- 3
x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="MkP")] <- 4

#So these are the comparisons we specifaclly want to make,
#CD41neg vs. pos
#FLHSC vs. CD41neg+CD41pos
#MkP vs. CD41neg+CD41pos
#FL HSC vs CD41neg (200916)

#CD41neg vs. pos
DARs = findDAR(
    obj=x.sp3.sample_peaks,
    input.mat="pmat",
    cluster.pos=3,
    cluster.neg=2,
    test.method="exactTest",
    bcv=0.1, #0.4 for human, 0.1 for mouse
    seed.use=10
  );

DARs$FDR = p.adjust(DARs$PValue, method="BH");
idy = which(DARs$FDR < 5e-2 & DARs$logFC > 0);
par(mfrow = c(1, 2));
plot(DARs$logCPM, DARs$logFC, 
    pch=19, cex=0.1, col="grey", 
    ylab="logFC", xlab="logCPM",
    main="CD41neg vs CD41pos"
  );
points(DARs$logCPM[idy], 
    DARs$logFC[idy], 
    pch=19, 
    cex=0.5, 
    col="red"
  );
abline(h = 0, lwd=1, lty=2);

if((x=length(idy)) < 2000L){
			PValues = DARs$PValue;
			PValues[DARs$logFC < 0] = 1;
			idy = order(PValues, decreasing=FALSE)[1:2000];
			rm(PValues); # free memory
}

covs = Matrix::rowSums(x.sp3.sample_peaks@pmat);
vals = Matrix::rowSums(x.sp3.sample_peaks@pmat[,idy]) / covs;
vals.zscore = (vals - mean(vals)) / sd(vals);
plotFeatureSingle(
    obj=x.sp3.sample_peaks,
    feature.value=vals.zscore,
    method="umap", 
    main="CD41neg vs CD41pos",
    point.size=0.1, 
    point.shape=19, 
    quantiles=c(0.2, 0.8)
  );

idyCD41neg <- idy #89 differential peaks
#89 is way too few so base it on the top 2000


#Can I plot the vals.zscore distribution here

df.vals.zscore_neg_vs_pos <- cbind( as.data.frame(vals.zscore), as.data.frame(x.sp3.sample_peaks@sample) )
colnames(df.vals.zscore_neg_vs_pos)<- c("vals.zscore", "sample")

library(plyr)
mu <- ddply(df.vals.zscore_neg_vs_pos, "sample", summarise, grp.mean=mean(vals.zscore))

CD41negvspos_zscore_density_plot <- ggplot(df.vals.zscore_neg_vs_pos, aes(vals.zscore, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() ) 



ggsave(plot = CD41negvspos_zscore_density_plot, filename = "CD41negvspos_zscore_density_plot_downscaled.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 10, height = 10)


#CD41pos vs. posneg
DARs = findDAR(
    obj=x.sp3.sample_peaks,
    input.mat="pmat",
    cluster.pos=2,
    cluster.neg=3,
    test.method="exactTest",
    bcv=0.1, #0.4 for human, 0.1 for mouse
    seed.use=10
  );

DARs$FDR = p.adjust(DARs$PValue, method="BH");
idy = which(DARs$FDR < 5e-2 & DARs$logFC > 0);


par(mfrow = c(1, 2));
plot(DARs$logCPM, DARs$logFC, 
    pch=19, cex=0.1, col="grey", 
    ylab="logFC", xlab="logCPM",
    main="CD41pos vs CD41neg"
  );
points(DARs$logCPM[idy], 
    DARs$logFC[idy], 
    pch=19, 
    cex=0.5, 
    col="red"
  );
abline(h = 0, lwd=1, lty=2);

#Take out the top 500 peaks for CD41pos
if((x=length(idy)) < 2000L){
			PValues = DARs$PValue;
			PValues[DARs$logFC < 0] = 1;
			idy = order(PValues, decreasing=FALSE)[1:2000];
			rm(PValues); # free memory
}

covs = Matrix::rowSums(x.sp3.sample_peaks@pmat);
vals = Matrix::rowSums(x.sp3.sample_peaks@pmat[,idy]) / covs;
vals.zscore = (vals - mean(vals)) / sd(vals);
plotFeatureSingle(
    obj=x.sp3.sample_peaks,
    feature.value=vals.zscore,
    method="umap", 
    main="CD41pos vs CD41neg",
    point.size=0.1, 
    point.shape=19, 
    quantiles=c(0.2, 0.8)
  );

#Can I plot the vals.zscore distribution here

df.vals.zscore_pos_vs_neg <- cbind( as.data.frame(vals.zscore), as.data.frame(x.sp3.sample_peaks@sample) )
colnames(df.vals.zscore_pos_vs_neg)<- c("vals.zscore", "sample")

library(plyr)
mu <- ddply(df.vals.zscore_pos_vs_neg, "sample", summarise, grp.mean=mean(vals.zscore))

CD41posvsneg_zscore_density_plot <- ggplot(df.vals.zscore_pos_vs_neg, aes(vals.zscore, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() ) 


ggsave(plot = CD41posvsneg_zscore_density_plot, filename = "CD41posvsneg_zscore_density_plot_downscaled.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 10, height = 10)

idyCD41pos <- idy #3164 DARs
#idyCD41pos_500 <- idy_500

#DARs need to be called on integers in the cluster object so over write it with sample
x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="FL_HSC")] <- 1
x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="CD41pos")] <- 2
x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="CD41neg")] <- 2
x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="MkP")] <- 4

#FLHSC vs ABMHSC
DARs = findDAR(
    obj=x.sp3.sample_peaks,
    input.mat="pmat",
    cluster.pos=1,
    cluster.neg=2,
    test.method="exactTest",
    bcv=0.1, #0.4 for human, 0.1 for mouse
    seed.use=10
  );

DARs$FDR = p.adjust(DARs$PValue, method="BH");
idy = which(DARs$FDR < 5e-2 & DARs$logFC > 0);
par(mfrow = c(1, 2));
plot(DARs$logCPM, DARs$logFC, 
    pch=19, cex=0.1, col="grey", 
    ylab="logFC", xlab="logCPM",
    main="FL HSC vs ABM HSC"
  );
points(DARs$logCPM[idy], 
    DARs$logFC[idy], 
    pch=19, 
    cex=0.5, 
    col="red"
  );
abline(h = 0, lwd=1, lty=2);
covs = Matrix::rowSums(x.sp3.sample_peaks@pmat);
vals = Matrix::rowSums(x.sp3.sample_peaks@pmat[,idy]) / covs;
vals.zscore = (vals - mean(vals)) / sd(vals);
plotFeatureSingle(
    obj=x.sp3.sample_peaks,
    feature.value=vals.zscore,
    method="umap", 
    main="FL HSC vs ABM HSC",
    point.size=0.1, 
    point.shape=19, 
    quantiles=c(0.01, 0.99)
  );

idyFLHSC <- idy #37472 DARs

#ABMHSC vs FLHSC
DARs = findDAR(
    obj=x.sp3.sample_peaks,
    input.mat="pmat",
    cluster.pos=2,
    cluster.neg=1,
    test.method="exactTest",
    bcv=0.1, #0.4 for human, 0.1 for mouse
    seed.use=10
  );

DARs$FDR = p.adjust(DARs$PValue, method="BH");
idy = which(DARs$FDR < 5e-2 & DARs$logFC > 0);
par(mfrow = c(1, 2));
plot(DARs$logCPM, DARs$logFC, 
    pch=19, cex=0.1, col="grey", 
    ylab="logFC", xlab="logCPM",
    main="ABM HSC vs FL HSC"
  );
points(DARs$logCPM[idy], 
    DARs$logFC[idy], 
    pch=19, 
    cex=0.5, 
    col="red"
  );
abline(h = 0, lwd=1, lty=2);
covs = Matrix::rowSums(x.sp3.sample_peaks@pmat);
vals = Matrix::rowSums(x.sp3.sample_peaks@pmat[,idy]) / covs;
vals.zscore = (vals - mean(vals)) / sd(vals);
plotFeatureSingle(
    obj=x.sp3.sample_peaks,
    feature.value=vals.zscore,
    method="umap", 
    main="ABM HSC vs FL HSC",
    point.size=0.1, 
    point.shape=19, 
    quantiles=c(0.01, 0.99)
  );

idyABMHSCvsFL <- idy #26185 DARs


#MkP vs ABMHSC DARs
DARs = findDAR(
    obj=x.sp3.sample_peaks,
    input.mat="pmat",
    cluster.pos=4,
    cluster.neg=2,
    test.method="exactTest",
    bcv=0.1, #0.4 for human, 0.1 for mouse
    seed.use=10
  );

DARs$FDR = p.adjust(DARs$PValue, method="BH");
idy = which(DARs$FDR < 5e-2 & DARs$logFC > 0);
par(mfrow = c(1, 2));
plot(DARs$logCPM, DARs$logFC, 
    pch=19, cex=0.1, col="grey", 
    ylab="logFC", xlab="logCPM",
    main="MkP vs ABM HSC"
  );
points(DARs$logCPM[idy], 
    DARs$logFC[idy], 
    pch=19, 
    cex=0.5, 
    col="red"
  );
abline(h = 0, lwd=1, lty=2);
covs = Matrix::rowSums(x.sp3.sample_peaks@pmat);
vals = Matrix::rowSums(x.sp3.sample_peaks@pmat[,idy]) / covs;
vals.zscore = (vals - mean(vals)) / sd(vals);
plotFeatureSingle(
    obj=x.sp3.sample_peaks,
    feature.value=vals.zscore,
    method="umap", 
    main="MkP vs ABM HSCC",
    point.size=0.1, 
    point.shape=19, 
    quantiles=c(0.01, 0.99)
  );

idyMkP <- idy #42005 DARs

#MkP vs ABMHSC DARs
DARs = findDAR(
    obj=x.sp3.sample_peaks,
    input.mat="pmat",
    cluster.pos=2,
    cluster.neg=4,
    test.method="exactTest",
    bcv=0.1, #0.4 for human, 0.1 for mouse
    seed.use=10
  );

DARs$FDR = p.adjust(DARs$PValue, method="BH");
idy = which(DARs$FDR < 5e-2 & DARs$logFC > 0);
par(mfrow = c(1, 2));
plot(DARs$logCPM, DARs$logFC, 
    pch=19, cex=0.1, col="grey", 
    ylab="logFC", xlab="logCPM",
    main="ABM HSC vs MkP"
  );
points(DARs$logCPM[idy], 
    DARs$logFC[idy], 
    pch=19, 
    cex=0.5, 
    col="red"
  );
abline(h = 0, lwd=1, lty=2);
covs = Matrix::rowSums(x.sp3.sample_peaks@pmat);
vals = Matrix::rowSums(x.sp3.sample_peaks@pmat[,idy]) / covs;
vals.zscore = (vals - mean(vals)) / sd(vals);
plotFeatureSingle(
    obj=x.sp3.sample_peaks,
    feature.value=vals.zscore,
    method="umap", 
    main="ABM HSC vs MkP",
    point.size=0.1, 
    point.shape=19, 
    quantiles=c(0.01, 0.99)
  );

idyABMHSCvsMkP <- idy #43053 DARs

###Added 200916
#DARs need to be called on integers in the cluster object so over write it with sample
x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="FL_HSC")] <- 1
x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="CD41pos")] <- 2
x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="CD41neg")] <- 3
x.sp3.sample_peaks@cluster[which(x.sp3.sample_peaks@sample=="MkP")] <- 4


#CD41neg vs. FLHSC
DARs = findDAR(
    obj=x.sp3.sample_peaks,
    input.mat="pmat",
    cluster.pos=3,
    cluster.neg=1,
    test.method="exactTest",
    bcv=0.1, #0.4 for human, 0.1 for mouse
    seed.use=10
  );

DARs$FDR = p.adjust(DARs$PValue, method="BH");
idy = which(DARs$FDR < 5e-2 & DARs$logFC > 0); #17859 diff peaks
par(mfrow = c(1, 2));
plot(DARs$logCPM, DARs$logFC, 
    pch=19, cex=0.1, col="grey", 
    ylab="logFC", xlab="logCPM",
    main="CD41neg vs FLHSC"
  );
points(DARs$logCPM[idy], 
    DARs$logFC[idy], 
    pch=19, 
    cex=0.5, 
    col="red"
  );
abline(h = 0, lwd=1, lty=2);

if((x=length(idy)) < 2000L){
			PValues = DARs$PValue;
			PValues[DARs$logFC < 0] = 1;
			idy = order(PValues, decreasing=FALSE)[1:2000];
			rm(PValues); # free memory
}

covs = Matrix::rowSums(x.sp3.sample_peaks@pmat);
vals = Matrix::rowSums(x.sp3.sample_peaks@pmat[,idy]) / covs;
vals.zscore = (vals - mean(vals)) / sd(vals);
plotFeatureSingle(
    obj=x.sp3.sample_peaks,
    feature.value=vals.zscore,
    method="umap", 
    main="CD41neg vs FL HSC",
    point.size=0.1, 
    point.shape=19, 
    quantiles=c(0.2, 0.8)
  );

idyCD41negvsFLHSC <- idy  #17859 diff peaks


#Can I plot the vals.zscore distribution here

df.vals.zscore_neg_vs_FL_HSC <- cbind( as.data.frame(vals.zscore), as.data.frame(x.sp3.sample_peaks@sample) )
colnames(df.vals.zscore_neg_vs_FL_HSC)<- c("vals.zscore", "sample")

library(plyr)
mu <- ddply(df.vals.zscore_neg_vs_FL_HSC, "sample", summarise, grp.mean=mean(vals.zscore))

CD41negvsFLHSC_zscore_density_plot <- ggplot(df.vals.zscore_neg_vs_FL_HSC, aes(vals.zscore, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() ) 



ggsave(plot = CD41negvsFLHSC_zscore_density_plot, filename = "CD41negvsFLHSC_zscore_density_plot.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/downscaled_DARs_peak_files/", units = c("cm"), width = 10, height = 10)

#FLHSC vs. CD41neg
DARs = findDAR(
    obj=x.sp3.sample_peaks,
    input.mat="pmat",
    cluster.pos=1,
    cluster.neg=3,
    test.method="exactTest",
    bcv=0.1, #0.4 for human, 0.1 for mouse
    seed.use=10
  );

DARs$FDR = p.adjust(DARs$PValue, method="BH");
idy = which(DARs$FDR < 5e-2 & DARs$logFC > 0); #39224 diff peaks
par(mfrow = c(1, 2));
plot(DARs$logCPM, DARs$logFC, 
    pch=19, cex=0.1, col="grey", 
    ylab="logFC", xlab="logCPM",
    main="CD41neg vs FLHSC"
  );
points(DARs$logCPM[idy], 
    DARs$logFC[idy], 
    pch=19, 
    cex=0.5, 
    col="red"
  );
abline(h = 0, lwd=1, lty=2);

if((x=length(idy)) < 2000L){
			PValues = DARs$PValue;
			PValues[DARs$logFC < 0] = 1;
			idy = order(PValues, decreasing=FALSE)[1:2000];
			rm(PValues); # free memory
}

covs = Matrix::rowSums(x.sp3.sample_peaks@pmat);
vals = Matrix::rowSums(x.sp3.sample_peaks@pmat[,idy]) / covs;
vals.zscore = (vals - mean(vals)) / sd(vals);
plotFeatureSingle(
    obj=x.sp3.sample_peaks,
    feature.value=vals.zscore,
    method="umap", 
    main="FLHSC vs CD41neg",
    point.size=0.1, 
    point.shape=19, 
    quantiles=c(0.2, 0.8)
  );

idyFLHSCvsCD41neg <- idy  #39224 diff peaks


#Can I plot the vals.zscore distribution here

df.vals.zscore_FL_HSC_vs_CD41neg <- cbind( as.data.frame(vals.zscore), as.data.frame(x.sp3.sample_peaks@sample) )
colnames(df.vals.zscore_FL_HSC_vs_CD41neg)<- c("vals.zscore", "sample")

library(plyr)
mu <- ddply(df.vals.zscore_FL_HSC_vs_CD41neg, "sample", summarise, grp.mean=mean(vals.zscore))

FLHSCvsCD41neg_zscore_density_plot <- ggplot(df.vals.zscore_FL_HSC_vs_CD41neg, aes(vals.zscore, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() ) 



ggsave(plot = FLHSCvsCD41neg_zscore_density_plot, filename = "FLHSCvsCD41neg_zscore_density_plot.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/downscaled_DARs_peak_files/", units = c("cm"), width = 10, height = 10)


#CD41pos vs. FLHSC
DARs = findDAR(
    obj=x.sp3.sample_peaks,
    input.mat="pmat",
    cluster.pos=2,
    cluster.neg=1,
    test.method="exactTest",
    bcv=0.1, #0.4 for human, 0.1 for mouse
    seed.use=10
  );

DARs$FDR = p.adjust(DARs$PValue, method="BH");
idy = which(DARs$FDR < 5e-2 & DARs$logFC > 0); #17859 diff peaks
par(mfrow = c(1, 2));
plot(DARs$logCPM, DARs$logFC, 
    pch=19, cex=0.1, col="grey", 
    ylab="logFC", xlab="logCPM",
    main="CD41pos vs FLHSC"
  );
points(DARs$logCPM[idy], 
    DARs$logFC[idy], 
    pch=19, 
    cex=0.5, 
    col="red"
  );
abline(h = 0, lwd=1, lty=2);

if((x=length(idy)) < 2000L){
			PValues = DARs$PValue;
			PValues[DARs$logFC < 0] = 1;
			idy = order(PValues, decreasing=FALSE)[1:2000];
			rm(PValues); # free memory
}

covs = Matrix::rowSums(x.sp3.sample_peaks@pmat);
vals = Matrix::rowSums(x.sp3.sample_peaks@pmat[,idy]) / covs;
vals.zscore = (vals - mean(vals)) / sd(vals);
plotFeatureSingle(
    obj=x.sp3.sample_peaks,
    feature.value=vals.zscore,
    method="umap", 
    main="CD41pos vs FL HSC",
    point.size=0.1, 
    point.shape=19, 
    quantiles=c(0.2, 0.8)
  );

idyCD41posvsFLHSC <- idy  #19134 diff peaks


#Can I plot the vals.zscore distribution here

df.vals.zscore_pos_vs_FL_HSC <- cbind( as.data.frame(vals.zscore), as.data.frame(x.sp3.sample_peaks@sample) )
colnames(df.vals.zscore_pos_vs_FL_HSC)<- c("vals.zscore", "sample")

library(plyr)
mu <- ddply(df.vals.zscore_pos_vs_FL_HSC, "sample", summarise, grp.mean=mean(vals.zscore))

CD41posvsFLHSC_zscore_density_plot <- ggplot(df.vals.zscore_pos_vs_FL_HSC, aes(vals.zscore, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() ) 



ggsave(plot = CD41posvsFLHSC_zscore_density_plot, filename = "CD41posvsFLHSC_zscore_density_plot.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/downscaled_DARs_peak_files/", units = c("cm"), width = 10, height = 10)


#FLHSC vs. CD41pos
DARs = findDAR(
    obj=x.sp3.sample_peaks,
    input.mat="pmat",
    cluster.pos=1,
    cluster.neg=2,
    test.method="exactTest",
    bcv=0.1, #0.4 for human, 0.1 for mouse
    seed.use=10
  );

DARs$FDR = p.adjust(DARs$PValue, method="BH");
idy = which(DARs$FDR < 5e-2 & DARs$logFC > 0); #29381 diff peaks
par(mfrow = c(1, 2));
plot(DARs$logCPM, DARs$logFC, 
    pch=19, cex=0.1, col="grey", 
    ylab="logFC", xlab="logCPM",
    main="CD41pos vs FLHSC"
  );
points(DARs$logCPM[idy], 
    DARs$logFC[idy], 
    pch=19, 
    cex=0.5, 
    col="red"
  );
abline(h = 0, lwd=1, lty=2);

if((x=length(idy)) < 2000L){
			PValues = DARs$PValue;
			PValues[DARs$logFC < 0] = 1;
			idy = order(PValues, decreasing=FALSE)[1:2000];
			rm(PValues); # free memory
}

covs = Matrix::rowSums(x.sp3.sample_peaks@pmat);
vals = Matrix::rowSums(x.sp3.sample_peaks@pmat[,idy]) / covs;
vals.zscore = (vals - mean(vals)) / sd(vals);
plotFeatureSingle(
    obj=x.sp3.sample_peaks,
    feature.value=vals.zscore,
    method="umap", 
    main="FL HSC vs DC41pos",
    point.size=0.1, 
    point.shape=19, 
    quantiles=c(0.2, 0.8)
  );

idyFLHSCvsCD41pos <- idy  #29381 diff peaks


#Can I plot the vals.zscore distribution here

df.vals.zscore_FL_HSC_vs_CD41pos <- cbind( as.data.frame(vals.zscore), as.data.frame(x.sp3.sample_peaks@sample) )
colnames(df.vals.zscore_FL_HSC_vs_CD41pos)<- c("vals.zscore", "sample")

library(plyr)
mu <- ddply(df.vals.zscore_FL_HSC_vs_CD41pos, "sample", summarise, grp.mean=mean(vals.zscore))

FLHSCvsCD41pos_zscore_density_plot <- ggplot(df.vals.zscore_FL_HSC_vs_CD41pos, aes(vals.zscore, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() ) 



ggsave(plot = FLHSCvsCD41pos_zscore_density_plot, filename = "FLHSCvsCD41pos_zscore_density_plot.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/downscaled_DARs_peak_files/", units = c("cm"), width = 10, height = 10)


# CD41pos vs MkP
DARs = findDAR(
    obj=x.sp3.sample_peaks,
    input.mat="pmat",
    cluster.pos=2,
    cluster.neg=4,
    test.method="exactTest",
    bcv=0.1, #0.4 for human, 0.1 for mouse
    seed.use=10
  );

DARs$FDR = p.adjust(DARs$PValue, method="BH");
idy = which(DARs$FDR < 5e-2 & DARs$logFC > 0); #26273 diff peaks
par(mfrow = c(1, 2));
plot(DARs$logCPM, DARs$logFC, 
    pch=19, cex=0.1, col="grey", 
    ylab="logFC", xlab="logCPM",
    main="CD41pos vs FLHSC"
  );
points(DARs$logCPM[idy], 
    DARs$logFC[idy], 
    pch=19, 
    cex=0.5, 
    col="red"
  );
abline(h = 0, lwd=1, lty=2);

if((x=length(idy)) < 2000L){
			PValues = DARs$PValue;
			PValues[DARs$logFC < 0] = 1;
			idy = order(PValues, decreasing=FALSE)[1:2000];
			rm(PValues); # free memory
}

covs = Matrix::rowSums(x.sp3.sample_peaks@pmat);
vals = Matrix::rowSums(x.sp3.sample_peaks@pmat[,idy]) / covs;
vals.zscore = (vals - mean(vals)) / sd(vals);
plotFeatureSingle(
    obj=x.sp3.sample_peaks,
    feature.value=vals.zscore,
    method="umap", 
    main="CD41pos vs MkP",
    point.size=0.1, 
    point.shape=19, 
    quantiles=c(0.2, 0.8)
  );

idyCD41posvsMkP <- idy  #26273 diff peaks


#Can I plot the vals.zscore distribution here

df.vals.zscore_CD41pos_vs_MkP <- cbind( as.data.frame(vals.zscore), as.data.frame(x.sp3.sample_peaks@sample) )
colnames(df.vals.zscore_CD41pos_vs_MkP)<- c("vals.zscore", "sample")

library(plyr)
mu <- ddply(df.vals.zscore_CD41pos_vs_MkP, "sample", summarise, grp.mean=mean(vals.zscore))

CD41posvsMkP_zscore_density_plot <- ggplot(df.vals.zscore_CD41pos_vs_MkP, aes(vals.zscore, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() ) 



ggsave(plot = CD41posvsMkP_zscore_density_plot, filename = "CD41posvsMkP_zscore_density_plot.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/downscaled_DARs_peak_files/", units = c("cm"), width = 10, height = 10)


#  MkP vs CD41pos
DARs = findDAR(
    obj=x.sp3.sample_peaks,
    input.mat="pmat",
    cluster.pos=4,
    cluster.neg=2,
    test.method="exactTest",
    bcv=0.1, #0.4 for human, 0.1 for mouse
    seed.use=10
  );

DARs$FDR = p.adjust(DARs$PValue, method="BH");
idy = which(DARs$FDR < 5e-2 & DARs$logFC > 0); #24628 diff peaks
par(mfrow = c(1, 2));
plot(DARs$logCPM, DARs$logFC, 
    pch=19, cex=0.1, col="grey", 
    ylab="logFC", xlab="logCPM",
    main="CD41pos vs FLHSC"
  );
points(DARs$logCPM[idy], 
    DARs$logFC[idy], 
    pch=19, 
    cex=0.5, 
    col="red"
  );
abline(h = 0, lwd=1, lty=2);

if((x=length(idy)) < 2000L){
			PValues = DARs$PValue;
			PValues[DARs$logFC < 0] = 1;
			idy = order(PValues, decreasing=FALSE)[1:2000];
			rm(PValues); # free memory
}

covs = Matrix::rowSums(x.sp3.sample_peaks@pmat);
vals = Matrix::rowSums(x.sp3.sample_peaks@pmat[,idy]) / covs;
vals.zscore = (vals - mean(vals)) / sd(vals);
plotFeatureSingle(
    obj=x.sp3.sample_peaks,
    feature.value=vals.zscore,
    method="umap", 
    main="MkP vs CD41pos",
    point.size=0.1, 
    point.shape=19, 
    quantiles=c(0.2, 0.8)
  );

idyMkPvsCD41pos <- idy  #24628 diff peaks


#Can I plot the vals.zscore distribution here

df.vals.zscore_MkP_vs_CD41pos <- cbind( as.data.frame(vals.zscore), as.data.frame(x.sp3.sample_peaks@sample) )
colnames(df.vals.zscore_MkP_vs_CD41pos)<- c("vals.zscore", "sample")

library(plyr)
mu <- ddply(df.vals.zscore_MkP_vs_CD41pos, "sample", summarise, grp.mean=mean(vals.zscore))

MkPvsCD41pos_zscore_density_plot <- ggplot(df.vals.zscore_MkP_vs_CD41pos, aes(vals.zscore, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() ) 



ggsave(plot = MkPvsCD41pos_zscore_density_plot, filename = "MkPvsCD41pos_zscore_density_plot.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/downscaled_DARs_peak_files/", units = c("cm"), width = 10, height = 10)
###




#Create a list with all the DARs positions
idy.ls <- list(idyFLHSC, idyABMHSCvsFL, idyCD41neg, idyCD41pos, idyMkP, idyABMHSCvsMkP, idyCD41negvsFLHSC, idyFLHSCvsCD41neg, idyCD41posvsFLHSC, idyFLHSCvsCD41pos, idyCD41posvsMkP, idyMkPvsCD41pos)

#Export these differential peak lists for a Homer based peak analysis
#idyFLHSC
write.table(x = peaks.df[idy.ls[[1]],], file = "FLHSCvsABMHSC_DARs.bed", sep = "\t", row.names = F, col.names = F, quote = F)
#idyABMHSCvsFL
write.table(x = peaks.df[idy.ls[[2]],], file = "ABMHSCvsFLHSC_DARs.bed", sep = "\t", row.names = F, col.names = F, quote = F)
#idyCD41neg
write.table(x = peaks.df[idy.ls[[3]],], file = "CD41negvsCD41pos_DARs.bed", sep = "\t", row.names = F, col.names = F, quote = F)
#idyCD41pos
write.table(x = peaks.df[idy.ls[[4]],], file = "CD41posvsCD41neg_DARs.bed", sep = "\t", row.names = F, col.names = F, quote = F)
#idyMkP
write.table(x = peaks.df[idy.ls[[5]],], file = "MkPvsABMHSC_DARs.bed", sep = "\t", row.names = F, col.names = F, quote = F)
#idyABMHSCvsMkP
write.table(x = peaks.df[idy.ls[[6]],], file = "ABMHSCvsMkP_DARs.bed", sep = "\t", row.names = F, col.names = F, quote = F)
#idyCD41pos_top500
write.table(x = peaks.df[idy_500,], file = "CD41posvsCD41neg_DARs_top500.bed", sep = "\t", row.names = F, col.names = F, quote = F)
#idyCD41negvsFLHSC
write.table(x = peaks.df[idy.ls[[7]],], file = "CD41negvsFLHSC_DARs.bed", sep = "\t", row.names = F, col.names = F, quote = F)
#idyFLHSCvsCD41neg
write.table(x = peaks.df[idy.ls[[8]],], file = "FLHSCvsCD41neg_DARs.bed", sep = "\t", row.names = F, col.names = F, quote = F)
#idyCD41posvsFLHSC
write.table(x = peaks.df[idy.ls[[9]],], file = "CD41posvsFLHSC_DARs.bed", sep = "\t", row.names = F, col.names = F, quote = F)
#idyFLHSCvsCD41pos
write.table(x = peaks.df[idy.ls[[10]],], file = "FLHSCvsCD41pos_DARs.bed", sep = "\t", row.names = F, col.names = F, quote = F)
#idyCD41posvsFLHSC
write.table(x = peaks.df[idy.ls[[11]],], file = "CD41posvsMkP_DARs.bed", sep = "\t", row.names = F, col.names = F, quote = F)
#idyFLHSCvsCD41pos
write.table(x = peaks.df[idy.ls[[12]],], file = "MkPvsCD41pos_DARs.bed", sep = "\t", row.names = F, col.names = F, quote = F)
#Restore the clusters
x.sp3.sample_peaks@cluster <- x.sp3.cluster


#Create an object with just the diff peaks that later will be used for motif enrichment.
setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/motifs_sample_based_DARs_from_sample_peaks")
library(SnapATAC)
library(GenomicRanges)
peaks.names = system("ls | grep bed", intern=TRUE)

#> peaks.names
#[1] "ABMHSCvsFLHSC_DARs.bed"    "ABMHSCvsMkP_DARs.bed"
#[3] "CD41negvsCD41pos_DARs.bed" "CD41posvsCD41neg_DARs.bed"
#[5] "FLHSCvsABMHSC_DARs.bed"    "MkPvsABMHSC_DARs.bed"

peak.gr.ls = lapply(peaks.names, function(x){
    peak.df = read.table(x)
    GRanges(peak.df[,1], IRanges(peak.df[,2], peak.df[,3]))
  })

peak.gr = reduce(Reduce(c, peak.gr.ls));
peak.gr

#Create a cell-by-peak matrix
 #Step below is dependent on Genomic Ranges.
x.sp3.DAR <- x.sp3.sample_peaks
rmPmatFromSnap(x.sp3.DAR)
x.sp3.DAR = createPmat(x.sp3.DAR, peak.gr,  do.par = TRUE, num.cores = 3)
x.sp3.DAR = makeBinary(x.sp3.DAR, mat="pmat")


```

```{r chromVAR_motif_discovery, echo=FALSE}
library(SnapATAC)
library(chromVAR);
library(motifmatchr);
library(SummarizedExperiment);
library(BSgenome.Mmusculus.UCSC.mm10);


#To get a complete list. Run this against both mouse and human and combine them in some way.
x.sp3.sample_peaks_mmat_mouse@mmat = runChromVAR(
    obj=x.sp3.sample_peaks_mmat_mouse,
    input.mat="pmat",
    genome=BSgenome.Mmusculus.UCSC.mm10,
    min.count=10,
    
    species="Mus musculus"
  );

x.sp3.sample_peaks_mmat_human@mmat = runChromVAR(
    obj=x.sp3.sample_peaks_mmat_human,
    input.mat="pmat",
    genome=BSgenome.Mmusculus.UCSC.mm10,
    min.count=10,
    species="Homo sapiens"
  );

#Combine the two mmats and remove the originals:
x.sp3.sample_peaks_mmat_combined <- x.sp3.sample_peaks
x.sp3.sample_peaks_mmat_mouse <- x.sp3.sample_peaks
x.sp3.sample_peaks_mmat_human <- x.sp3.sample_peaks
x.sp3.sample_peaks_mmat_combined@mmat <- cbind(x.sp3.sample_peaks_mmat_mouse@mmat,x.sp3.sample_peaks_mmat_human@mmat)
rm(x.sp3.sample_peaks_mmat_mouse)
rm(x.sp3.sample_peaks_mmat_human)

#Get the names of the motifs
colnames(x.sp3.sample_peaks_mmat_combined@mmat)

plot_order_samples <- factor(c(rep(c('FL_HSC'), times=2403), rep(c('CD41neg'), times=2310), rep(c('CD41pos'), times=1123), rep(c('MkP'), times=2986)), levels=(c('FL_HSC', 'CD41neg', 'CD41pos', 'MkP')))

plot_order_clusters <- c(rep(c(1), times=1488),rep(c(2), times=860),rep(c(3), times=885),rep(c(4), times=1284),rep(c(5), times=773),rep(c(6), times=391),rep(c(7), times=1337),rep(c(8), times=150),rep(c(9), times=242),rep(c(10), times=305),rep(c(11), times=1108), levels = c(1:11))

motif_i = "MA0080.4_SPI1";
dat = data.frame(x=plot_order_samples, y=x.sp3.sample_peaks_mmat_combined@mmat[,motif_i]);

ggplot(dat, aes(x=x, y=y, fill=x)) + 
	theme_classic() +
	geom_violin(draw_quantiles = c(0.25, 0.5, 0.75)) + 
	xlab("Sample") +
	ylab("motif enrichment") + 
	ggtitle(motif_i) +
	theme(
		  axis.text.x = element_text(angle = 90, hjust = 1),
		  axis.ticks.x=element_blank(),
		  legend.position = "none"
   );

#Loop over all the motifs in the file.
for (i in 1:length(colnames(x.sp3.sample_peaks_mmat_combined@mmat)) ) {

  motif_i = colnames(x.sp3.sample_peaks_mmat_combined@mmat)[i];
  
  dat = data.frame(x=plot_order_samples, y=x.sp3.sample_peaks_mmat_combined@mmat[,motif_i]);

  ggplot(dat, aes(x=x, y=y, fill=x)) + 
	  theme_classic() +
	  geom_violin(draw_quantiles = c(0.25, 0.5, 0.75)) + 
	  xlab("Sample") +
	  ylab("motif enrichment") + 
	  ggtitle(motif_i) +
    theme(
		  axis.text.x = element_text(angle = 90, hjust = 1),
		  axis.ticks.x=element_blank(),
		  legend.position = "none"
   );

  ggsave(filename = paste(motif_i,"chromVAR_enrichment_per_sample.pdf",sep = "."), device = "pdf", dpi = 150, units = c("cm"), width = 10, height = 9)
}

#Plot a heatmap of the motifs over pseudotime_
pseudo.df <- cbind(as.data.frame(sce2$slingClusters), as.data.frame(sce2$slingPseudotime_1))
colnames(pseudo.df)<-c("sample", "pseudotime")

pseudotime_order <- pseudo.df[order(pseudo.df$pseudotime),]
pseudotime_order$index <- as.numeric(rownames(pseudotime_order))
pseudotime_order$cluster <- x.sp3.sample_peaks_mmat_combined@cluster[pseudotime_order$index]
pseudotime_order$sample <- x.sp3.sample_peaks_mmat_combined@sample[pseudotime_order$index]
#Bring out the motif_matrix
motif_matrix <-as.data.frame(t(x.sp3.sample_peaks_mmat_combined@mmat))
colnames(motif_matrix) <- 1:8822

#Plot it with pheatmap.
library(pheatmap)

pheatmap(motif_matrix[,c(pseudotime_order$index)], cluster_cols = F ,annotation_col = pseudotime_order[,c(1:2,4)], show_rownames=F, show_colnames = F)

#The motif matrix is a little big so take out the top 10% most varying motifs and ploit them
motif_matrix$variance = apply(motif_matrix[,1:8822],1,var)

motif_matrix <- motif_matrix[order(motif_matrix$variance, decreasing = T),]
motif_matrix$counter <- 1:514
motif_matrix_top100 <- subset(motif_matrix, motif_matrix$counter <= 100)
motif_matrix_top100 <- motif_matrix_top100[,1:8822]


#Plot top motifs as a heatmap along the pseudotime trajectory.
ann_colors = list(sample=c("FL_HSC"="red","CD41neg"="blue","CD41pos"="green","MkP"="goldenrod1"))

breakList <- seq(-3, 3, by = 0.1)
pheatmap(motif_matrix_top100[,c(pseudotime_order$index)], cluster_cols = F ,annotation_col = pseudotime_order[,c(1:2)], show_rownames=T, show_colnames = F,clustering_distance_rows = "correlation", clustering_method = "ward.D2", color = colorRampPalette(rev(brewer.pal(n = 11, name = "RdYlBu")))(length(breakList)), breaks = breakList, annotation_colors = ann_colors, scale = "row")

#Try tp 


#Plot some motifs over the umap
colnames(x.sp3.sample_peaks_mmat_combined@mmat)[211]<-"MA0140.2_GATA1_TAL1"

 plotFeatureSingle( obj=x.sp3.sample_peaks_mmat_combined,feature.value=x.sp3.sample_peaks_mmat_combined@mmat[,'MA0841.1_NFE2'], method="umap",  main='MA0841.1_NFE2', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8) )

 plotFeatureSingle( obj=x.sp3.sample_peaks_mmat_combined,feature.value=x.sp3.sample_peaks_mmat_combined@mmat[,'MA0035.3_Gata1'], method="umap",  main='MA0035.3_Gata1', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8) )
 
 plotFeatureSingle( obj=x.sp3.sample_peaks_mmat_combined,feature.value=x.sp3.sample_peaks_mmat_combined@mmat[,'MA0795.1_SMAD3'], method="umap",  main='MA0795.1_SMAD3', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8) )
 
 plotFeatureSingle( obj=x.sp3.sample_peaks_mmat_combined,feature.value=x.sp3.sample_peaks_mmat_combined@mmat[,'MA0080.4_SPI1'], method="umap",  main='MA0080.4_SPI1', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8) )

 plotFeatureSingle( obj=x.sp3.sample_peaks_mmat_combined,feature.value=x.sp3.sample_peaks_mmat_combined@mmat[,'MA0140.2_GATA1_TAL1'], method="umap",  main='MA0140.2_GATA1_TAL1', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8) )
 
 plotFeatureSingle( obj=x.sp3.sample_peaks_mmat_combined,feature.value=x.sp3.sample_peaks_mmat_combined@mmat[,'MA0002.2_RUNX1'], method="umap",  main='MA0002.2_RUNX1', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8))
 
  plotFeatureSingle( obj=x.sp3.sample_peaks_mmat_combined,feature.value=x.sp3.sample_peaks_mmat_combined@mmat[,'MA0594.1_Hoxa9'], method="umap",  main='MA0594.1_Hoxa9', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8))
  
   plotFeatureSingle( obj=x.sp3.sample_peaks_mmat_combined,feature.value=x.sp3.sample_peaks_mmat_combined@mmat[,'MA0102.3_CEBPA'], method="umap",  main='MA0102.3_CEBPA', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8))
   
    plotFeatureSingle( obj=x.sp3.sample_peaks_mmat_combined,feature.value=x.sp3.sample_peaks_mmat_combined@mmat[,'MA0139.1_CTCF'], method="umap",  main='MA0139.1_CTCF', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8))
    
    plotFeatureSingle( obj=x.sp3.sample_peaks_mmat_combined,feature.value=x.sp3.sample_peaks_mmat_combined@mmat[,'MA0652.1_IRF8'], method="umap",  main='MA0652.1_IRF8', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8))

    plotFeatureSingle( obj=x.sp3.sample_peaks_mmat_combined,feature.value=x.sp3.sample_peaks_mmat_combined@mmat[,'MA0050.2_IRF1'], method="umap",  main='MA0050.2_IRF1', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8))
    
    plotFeatureSingle( obj=x.sp3.sample_peaks_mmat_combined,feature.value=x.sp3.sample_peaks_mmat_combined@mmat[,'MA0036.2_GATA2'], method="umap",  main='MA0036.2_GATA2', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8))
    

#####################################################################################################################
###########################Repeat the above analysis with the DAR regions only.
####################################################################################################################    
#To get a complete list. Run this against both mouse and human and combine them in some way.
    
x.sp3.DAR_mouse <- x.sp3.DAR
    
x.sp3.DAR_mouse@mmat = runChromVAR(
    obj=x.sp3.DAR_mouse,
    input.mat="pmat",
    genome=BSgenome.Mmusculus.UCSC.mm10,
    min.count=10,
    species="Mus musculus"
  );

x.sp3.DAR_human <- x.sp3.DAR

x.sp3.DAR_human@mmat = runChromVAR(
    obj=x.sp3.DAR_human,
    input.mat="pmat",
    genome=BSgenome.Mmusculus.UCSC.mm10,
    min.count=10,
    species="Homo sapiens"
  );

#Combine the two mmats and remove the originals:
x.sp3.DAR_mmat_combined <- x.sp3.DAR
x.sp3.DAR_mmat_combined@mmat <- cbind(x.sp3.DAR_mouse@mmat,x.sp3.DAR_human@mmat)
rm(x.sp3.DAR_mouse)
rm(x.sp3.DAR_human)

pseudotime_order <- pseudo.df[order(pseudo.df$pseudotime),]
pseudotime_order$index <- as.numeric(rownames(pseudotime_order))
pseudotime_order$cluster <- x.sp3.DAR_mmat_combined@cluster[pseudotime_order$index]
pseudotime_order$sample <- x.sp3.DAR_mmat_combined@sample[pseudotime_order$index]
#Bring out the motif_matrix
motif_matrix <-as.data.frame(t(x.sp3.DAR_mmat_combined@mmat))
colnames(motif_matrix) <- 1:8823

#Plot it with pheatmap.
library(pheatmap)

pheatmap(motif_matrix[,c(pseudotime_order$index)], cluster_cols = F ,annotation_col = pseudotime_order[,c(1:2,4)], show_rownames=F, show_colnames = F)

#The motif matrix is a little big so take out the top 10% most varying motifs and ploit them
motif_matrix$variance = apply(motif_matrix[,1:8823],1,var)

motif_matrix <- motif_matrix[order(motif_matrix$variance, decreasing = T),]
motif_matrix$counter <- 1:514
motif_matrix_top100 <- subset(motif_matrix, motif_matrix$counter <= 100)
motif_matrix_top100 <- motif_matrix_top100[,1:8823]


#Plot top motifs as a heatmap along the pseudotime trajectory.
ann_colors = list(sample=c("FL_HSC"="red","CD41neg"="blue","CD41pos"="green","MkP"="goldenrod1"))

pheatmap(motif_matrix_top100[,c(pseudotime_order$index)], cluster_cols = F ,annotation_col = pseudotime_order[,c(1:2,4)], show_rownames=T, show_colnames = F,clustering_distance_rows = "correlation", clustering_method = "ward.D2", annotation_colors = ann_colors, scale = "column")

#Loop over all the motifs in the file.
for (i in 1:length(colnames(x.sp3.DAR_mmat_combined@mmat)) ) {

  motif_i = colnames(x.sp3.DAR_mmat_combined@mmat)[i];
  
  dat = data.frame(x=plot_order_samples, y=x.sp3.DAR_mmat_combined@mmat[,motif_i]);

  ggplot(dat, aes(x=x, y=y, fill=x)) + 
	  theme_classic() +
	  geom_violin(draw_quantiles = c(0.25, 0.5, 0.75)) + 
	  xlab("Sample") +
	  ylab("motif enrichment") + 
	  ggtitle(motif_i) +
    theme(
		  axis.text.x = element_text(angle = 90, hjust = 1),
		  axis.ticks.x=element_blank(),
		  legend.position = "none"
   );

  ggsave(filename = paste(motif_i,"chromVAR_enrichment_per_sample_DARs_only.pdf",sep = "."), device = "pdf", dpi = 150, units = c("cm"), width = 10, height = 9)
}
#Plot some motifs over the umap
colnames(x.sp3.DAR_mmat_combined@mmat)[211]<-"MA0140.2_GATA1_TAL1"

plotFeatureSingle( obj=x.sp3.DAR_mmat_combined,feature.value=x.sp3.DAR_mmat_combined@mmat[,'MA0841.1_NFE2'], method="umap",  main='MA0841.1_NFE2', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8) )

plotFeatureSingle( obj=x.sp3.DAR_mmat_combined,feature.value=x.sp3.DAR_mmat_combined@mmat[,'MA0035.3_Gata1'], method="umap",  main='MA0035.3_Gata1', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8) )

plotFeatureSingle( obj=x.sp3.DAR_mmat_combined,feature.value=x.sp3.DAR_mmat_combined@mmat[,'MA0795.1_SMAD3'], method="umap",  main='MA0795.1_SMAD3', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8) )

plotFeatureSingle( obj=x.sp3.DAR_mmat_combined,feature.value=x.sp3.DAR_mmat_combined@mmat[,'MA0080.4_SPI1'], method="umap",  main='MA0080.4_SPI1', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8) )

plotFeatureSingle( obj=x.sp3.DAR_mmat_combined,feature.value=x.sp3.DAR_mmat_combined@mmat[,'MA0140.2_GATA1_TAL1'], method="umap",  main='MA0140.2_GATA1_TAL1', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8) )

plotFeatureSingle( obj=x.sp3.DAR_mmat_combined,feature.value=x.sp3.DAR_mmat_combined@mmat[,'MA0002.2_RUNX1'], method="umap",  main='MA0002.2_RUNX1', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8))

plotFeatureSingle( obj=x.sp3.DAR_mmat_combined,feature.value=x.sp3.DAR_mmat_combined@mmat[,'MA0594.1_Hoxa9'], method="umap",  main='MA0594.1_Hoxa9', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8))

plotFeatureSingle( obj=x.sp3.DAR_mmat_combined,feature.value=x.sp3.DAR_mmat_combined@mmat[,'MA0102.3_CEBPA'], method="umap",  main='MA0102.3_CEBPA', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8))

plotFeatureSingle( obj=x.sp3.DAR_mmat_combined,feature.value=x.sp3.DAR_mmat_combined@mmat[,'MA0139.1_CTCF'], method="umap",  main='MA0139.1_CTCF', point.size=0.1,  point.shape=19 , quantiles=c(0.2, 0.8))

## DAR motif analysis ends here.
#################################################################################################
#################################################################################################
#################################################################################################
```





```{r visualization_of_external_peaks, echo=FALSE}

#Up FL_HSC
x.sp3.ext_peaks <- x.sp3.sample_peaks
rmPmatFromSnap(x.sp3.ext_peaks)

setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/diff_peak_from_bulk_atac/Up_FL_HSC/")
peaks.names = system("ls | grep bed", intern=TRUE)
peak.gr.ls = lapply(peaks.names, function(x){
    peak.df = read.table(x)
    GRanges(peak.df[,1], IRanges(peak.df[,2], peak.df[,3]))
  })

peak.gr = reduce(Reduce(c, peak.gr.ls));
peak.gr
#Create a cell-by-peak matrix
library(GenomicRanges) #Step below is dependent on Genomic Ranges.

x.sp3.ext_peaks = createPmat(x.sp3.ext_peaks, peak.gr,  do.par = TRUE, num.cores = 8)
#Lift out the matrix and normalize it with one of The Hill method from Signac. This comparison should be done in two ways. 1) as a ratio compared to all the peaks OR compute the overlap for each cell-type specific peak vs the enhancer list of interest.
#read in the function


source("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/RunTFIDF.signac.R")
peak.matrix_up_bulk_fl <- as(object = x.sp3.ext_peaks@pmat, Class = 'matrix')
peak.matrix <- as(object = x.sp3.sample_peaks@pmat, Class = 'matrix')
ncol(peak.matrix)
#Split the peak matrix into cell types and perform TIFD and z-scaling on them both:
#Create a hardcoded function for this. It applies Signac TIDF.norm with the Cusanovich method as default.

TIFD_zscore_norm <- function(mat, tfid_norm = 1){
 
  #Transpose and split the matrix
  mat <- t(mat)
  #Ugly split it into each cell type:
  mat_FL <- mat[,1:2403]
  mat_CD41neg <- mat[,2404:4713]
  mat_CD41pos <- mat[,4714:5836]
  mat_MkP <- mat[,5837:8822]
  
  library(Signac)
  library(Seurat)
  #FL
  mat_FL <- as.matrix(RunTFIDF(object =  mat_FL, method = tfid_norm) )
  print(ncol(mat_FL))
  print(mean(apply(mat_FL, 2, function(c)sum(c==0))))
  #mat_FL <- apply(mat_FL,2,scale)
  #CD41neg
  mat_CD41neg <- as.matrix(RunTFIDF(object =  mat_CD41neg, method = tfid_norm) )
  print(dim(mat_CD41neg))
  print(mean(apply(mat_CD41neg, 2, function(c)sum(c==0))))
  #mat_CD41neg <- apply(mat_CD41neg,2,scale)
  #CD41neg
  mat_CD41pos <- as.matrix(RunTFIDF(object =  mat_CD41pos, method = tfid_norm) )
  print(dim(mat_CD41pos))
  print(mean(apply(mat_CD41pos, 2, function(c)sum(c==0))))
  #mat_CD41pos <- apply(mat_CD41pos,2,scale)
  #MkP
  mat_MkP<- as.matrix(RunTFIDF(object =  mat_MkP, method = tfid_norm) )
  print(dim(mat_MkP))
  print(mean(apply(mat_MkP, 2, function(c)sum(c==0))))
  #mat_MkP <- apply(mat_MkP,2,scale)
  
  #Combine to one matrix
  #norm.mat <- apply(cbind(mat_FL,mat_CD41neg,mat_CD41pos,mat_MkP),1,scale)
  norm.mat <- cbind(mat_FL,mat_CD41neg,mat_CD41pos,mat_MkP)
  #Remove cols with NA
  #norm.mat <-  norm.mat[complete.cases(norm.mat), ]
  return(norm.mat)
}

#For additional peak matrix calculations see below:

norm.mat_up_fl <- t(TIFD_zscore_norm(peak.matrix_up_bulk_fl, tfid_norm = 1))
norm.mat_down_fl <- t(TIFD_zscore_norm(peak.matrix_down_bulk_fl, tfid_norm = 1))
norm.mat_full <- t(TIFD_zscore_norm(as(object = x.sp3.sample_peaks@pmat, Class = 'matrix'), tfid_norm = 1))
norm.mat.NFE2_MkP <-t(TIFD_zscore_norm(peak.matrix_NFE2_MkP, tfid_norm = 1)) 
norm.mat.NFE2_ery <- t(TIFD_zscore_norm(peak.matrix_NFE2_ery, tfid_norm = 1)) 
norm.mat.gata1_mk <- t(TIFD_zscore_norm(peak.matrix_gata1_mk, tfid_norm = 1)) 
norm.mat.gata1_ery <- t(TIFD_zscore_norm(peak.matrix_gata1_ery, tfid_norm = 1)) 
norm.mat.iMK_unique_AREs <- t(TIFD_zscore_norm(peak.matrix_iMK_unique_AREs, tfid_norm = 1)) 
norm.mat.ery_unique_AREs <- t(TIFD_zscore_norm(peak.matrix_ery_unique_AREs, tfid_norm = 1)) 

#Will not save the peak matrices or the normalized full peak matrix since they are huge.
TIFD_norm_peak.matrices = list(c(norm.mat_up_fl=norm.mat_up_fl, norm.mat_down_fl = norm.mat_down_fl, norm.mat.NFE2_MkP = norm.mat.NFE2_MkP, norm.mat.NFE2_ery = norm.mat.NFE2_ery, norm.mat.gata1_mk = norm.mat.gata1_mk, norm.mat.gata1_ery = norm.mat.gata1_ery, norm.mat.iMK_unique_AREs = norm.mat.iMK_unique_AREs, norm.mat.ery_unique_AREs = norm.mat.ery_unique_AREs))

norm.up_fl <- as.data.frame( apply(norm.mat_up_fl,1,sum, na.rm=T))
norm.down_fl <- as.data.frame( apply(norm.mat_down_fl,1,sum, na.rm=T))
norm.NFE2_MkP <- as.data.frame( apply(norm.mat.NFE2_MkP,1,sum, na.rm=T))
norm.NFE2_ery <- as.data.frame( apply(norm.mat.NFE2_ery,1,sum, na.rm=T))
norm.gata1_mk <- as.data.frame( apply(norm.mat.gata1_mk,1,sum, na.rm=T))
norm.gata1_ery <- as.data.frame( apply(norm.mat.gata1_ery,1,sum, na.rm=T))
norm.iMK_unique_AREs <- as.data.frame( apply(norm.mat.iMK_unique_AREs,1,sum, na.rm=T))
norm.ery_unique_AREs <- as.data.frame( apply(norm.mat.ery_unique_AREs,1,sum, na.rm=T))
norm.mat <- as.data.frame( apply(norm.mat_full,1,sum, na.rm=T))

#Make a list of the normalized accessibility per cell:
norm.cell_access <- list(norm.up_fl = norm.up_fl, norm.down_fl = norm.down_fl, norm.NFE2_MkP = norm.NFE2_MkP, norm.NFE2_ery = norm.NFE2_ery, norm.gata1_mk = norm.gata1_mk, norm.gata1_ery = norm.gata1_ery, norm.iMK_unique_AREs = norm.iMK_unique_AREs, norm.ery_unique_AREs =  norm.ery_unique_AREs)

up_enriched <- norm.up_fl/norm.mat
down_enriched <- norm.down_fl/norm.mat
NFE2_MkP_enriched <- norm.NFE2_MkP/norm.mat
NFE2_ery_enriched <- norm.NFE2_ery/norm.mat
gata1_mk_enriched<- norm.gata1_mk/norm.mat
gata1_ery_enriched <- norm.gata1_ery/norm.mat
iMK_unique_AREs_enriched <- norm.iMK_unique_AREs/norm.mat 
ery_unique_AREs_enriched <- norm.ery_unique_AREs/norm.mat 

#Make a list of the relative accessibility:
cell.rel_access <- list(up_enriched = up_enriched, down_enriched = down_enriched, NFE2_MkP_enriched = NFE2_MkP_enriched, NFE2_ery_enriched = NFE2_ery_enriched, gata1_mk_enriched = gata1_mk_enriched, gata1_ery_enriched = gata1_ery_enriched, iMK_unique_AREs_enriched = iMK_unique_AREs_enriched, ery_unique_AREs_enriched = ery_unique_AREs_enriched)


cell.rel_access_df <- cbind( as.data.frame(x.sp3.ext_peaks@sample), norm.up_fl, norm.down_fl, norm.NFE2_MkP, norm.NFE2_ery, norm.gata1_mk, norm.gata1_ery, norm.iMK_unique_AREs, norm.ery_unique_AREs, norm.mat, up_enriched, down_enriched, NFE2_MkP_enriched, NFE2_ery_enriched, gata1_mk_enriched, gata1_ery_enriched, iMK_unique_AREs_enriched, ery_unique_AREs_enriched)


colnames(cell.rel_access_df) <- c("sample", "norm.up_fl", "norm.down_fl", "norm.NFE2_MkP", "norm.NFE2_ery", "norm.gata1_mk", "norm.gata1_ery", "norm.iMK_unique_AREs", "norm.ery_unique_AREs", "norm.mat.full", "up_enriched", "down_enriched", "NFE2_MkP_enriched", "NFE2_ery_enriched", "gata1_mk_enriched", "gata1_ery_enriched", "iMK_unique_AREs_enriched", "ery_unique_AREs_enriched")

plot_order_samples <- factor(c(rep(c('FL_HSC'), times=2403), rep(c('CD41neg'), times=2310), rep(c('CD41pos'), times=1123), rep(c('MkP'), times=2986)), levels=(c('FL_HSC', 'CD41neg', 'CD41pos', 'MkP')))

#Up_enriched
ggplot()+geom_violin(data = cell.rel_access_df, aes( y = up_enriched, x = plot_order_samples, fill = sample),draw_quantiles = c(0.25, 0.5, 0.75)) + theme_bw() + ggtitle("Opening bulk FL HSC: relative accessibility accessibility") + xlab("Sample") + ylab("Peak subset/total peak accessibility") +  ylim(0.03,0.2) 

ggsave( filename = "Opening bulk FL HSC: relative accessibility.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)

library(plyr)
#Density plot
mu <- ddply(cell.rel_access_df, "sample", summarise, grp.mean=mean(up_enriched))

ggplot(cell.rel_access_df, aes(up_enriched, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() )  + 
 ggtitle("Opening bulk FL HSC: relative accessibility accessibility") + xlab("Peak subset/total peak accessibility") + ylab("Density") +  xlim(0.03,0.2) 

ggsave( filename = "Density - Opening bulk FL HSC: relative accessibility.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)


#Down enriched
ggplot()+geom_violin(data = test, aes( y = down_enriched, x = plot_order_samples, fill = sample),draw_quantiles = c(0.25, 0.5, 0.75)) + theme_bw() + ggtitle("Opening bulk ABM HSC: relative accessibility accessibility") + xlab("Sample") + ylab("Peak subset/total peak accessibility")  +  ylim(0.03,0.2)

ggsave( filename = "Opening bulk ABM HSC: relative accessibility.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)

#Density plot
mu <- ddply(cell.rel_access_df, "sample", summarise, grp.mean=mean(down_enriched))

ggplot(cell.rel_access_df, aes(down_enriched, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() )  + 
 ggtitle("Closing bulk FL HSC: relative accessibility accessibility") + xlab("Peak subset/total peak accessibility") + ylab("Density") +  xlim(0.03,0.2) 

ggsave( filename = "Density - Closing bulk FL HSC: relative accessibility.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)


#NFE2_MkP_enriched
ggplot()+geom_violin(data = test, aes( y = NFE2_MkP_enriched, x = plot_order_samples, fill = sample),draw_quantiles = c(0.25, 0.5, 0.75)) + theme_bw() + ggtitle("MkP unique NFE2 peaks: relative accessibility accessibility") + xlab("Sample") + ylab("Peak subset/total peak accessibility") + ylim(0.03,0.15)

ggsave( filename = "MkP unique NFE2 peaks.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)

#Density plot
mu <- ddply(cell.rel_access_df, "sample", summarise, grp.mean=mean(NFE2_MkP_enriched))

ggplot(cell.rel_access_df, aes(NFE2_MkP_enriched, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() )  + 
 ggtitle("MkP unique NFE2 peaks: relative accessibility accessibility") + xlab("Peak subset/total peak accessibility") + ylab("Density") +  xlim(0.03,0.12) 

ggsave( filename = "Density - MkP unique NFE2 peaks: relative accessibility accessibility.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)

#NFE2_ery_enriched
ggplot()+geom_violin(data = test, aes( y = NFE2_ery_enriched, x = plot_order_samples, fill = sample),draw_quantiles = c(0.25, 0.5, 0.75)) + theme_bw() + ggtitle("Ery unique NFE2 peaks: relative accessibility accessibility") + xlab("Sample") + ylab("Peak subset/total peak accessibility") 

ggsave( filename = "Ery unique NFE2 peaks.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)

#Density plot
mu <- ddply(cell.rel_access_df, "sample", summarise, grp.mean=mean(NFE2_ery_enriched))

ggplot(cell.rel_access_df, aes(NFE2_ery_enriched, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() )  + 
 ggtitle("Ery unique NFE2 peaks: relative accessibility accessibility") + xlab("Peak subset/total peak accessibility") + ylab("Density") +  xlim(0.03,0.12) 

ggsave( filename = "Density - Ery unique NFE2 peaks: relative accessibility accessibility.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)

#gata1_mk_enriched
ggplot()+geom_violin(data = test, aes( y = gata1_mk_enriched, x = plot_order_samples, fill = sample),draw_quantiles = c(0.25, 0.5, 0.75)) + theme_bw() + ggtitle("MkP unique Gata1 peaks: relative accessibility accessibility") + xlab("Sample") + ylab("Peak subset/total peak accessibility") + ylim(0.02,0.15)

ggsave( filename = "MkP unique Gata1 peaks.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)

#Density plot
mu <- ddply(cell.rel_access_df, "sample", summarise, grp.mean=mean(gata1_mk_enriched))

ggplot(cell.rel_access_df, aes(gata1_mk_enriched, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() )  + 
 ggtitle("MkP unique Gata1 peaks: relative accessibility accessibility") + xlab("Peak subset/total peak accessibility") + ylab("Density") +  xlim(0.02,0.15) 

ggsave( filename = "Density - MkP unique Gata1 peaks: relative accessibility accessibility.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)

#gata1_ery_enriched
ggplot()+geom_violin(data = test, aes( y = gata1_ery_enriched, x = plot_order_samples, fill = sample),draw_quantiles = c(0.25, 0.5, 0.75)) + theme_bw() + ggtitle("Ery unique Gata1 peaks: relative accessibility accessibility") + xlab("Sample") + ylab("Peak subset/total peak accessibility") + ylim(0.02,0.15)

ggsave( filename = "Ery unique Gata1 peaks.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)

#Density plots
mu <- ddply(cell.rel_access_df, "sample", summarise, grp.mean=mean(gata1_ery_enriched))

ggplot(cell.rel_access_df, aes(gata1_ery_enriched, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() )  + 
 ggtitle("Ery unique Gata1 peaks: relative accessibility accessibility") + xlab("Peak subset/total peak accessibility") + ylab("Density") +  xlim(0.02,0.15) 

ggsave( filename = "Density - Ery unique Gata1 peaks: relative accessibility accessibility.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)


#iMK_unique_AREs_enriched
ggplot()+geom_violin(data = test, aes( y = iMK_unique_AREs_enriched, x = plot_order_samples, fill = sample),draw_quantiles = c(0.25, 0.5, 0.75)) + theme_bw() + ggtitle("iMK unique AREs: relative accessibility accessibility") + xlab("Sample") + ylab("Peak subset/total peak accessibility") 

ggsave( filename = "iMK unique AREs.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)
#Density plot
mu <- ddply(cell.rel_access_df, "sample", summarise, grp.mean=mean(iMK_unique_AREs_enriched))

ggplot(cell.rel_access_df, aes(iMK_unique_AREs_enriched, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() )  + 
 ggtitle("iMK unique AREs: relative accessibility accessibility") + xlab("Peak subset/total peak accessibility") + ylab("Density") 

ggsave( filename = "Density - iMK unique AREs: relative accessibility accessibility.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)

#ery_unique_AREs_enriched
ggplot()+geom_violin(data = test, aes( y = ery_unique_AREs_enriched, x = plot_order_samples, fill = sample),draw_quantiles = c(0.25, 0.5, 0.75)) + theme_bw() + ggtitle("Ery unique AREs: relative accessibility accessibility") + xlab("Sample") + ylab("Peak subset/total peak accessibility") 

ggsave( filename = "Ery unique AREs.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)

#Density plot
mu <- ddply(cell.rel_access_df, "sample", summarise, grp.mean=mean(ery_unique_AREs_enriched))

ggplot(cell.rel_access_df, aes(ery_unique_AREs_enriched, fill=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") ) ) ) +  geom_density(alpha=0.4) + theme_bw() + theme(legend.position = "bottom") + geom_vline(data=mu, aes(xintercept=grp.mean, color=factor(sample, levels = c("FL_HSC", "CD41neg", "CD41pos", "MkP") )),linetype="dashed") + theme( legend.title = element_blank() )  + 
 ggtitle("Ery unique AREs: relative accessibility accessibility") + xlab("Peak subset/total peak accessibility") + ylab("Density") + xlim(c(0.05,0.12))

ggsave( filename = "Density - Ery unique AREs: relative accessibility accessibility.pdf", device = "pdf",path = "/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/", units = c("cm"), width = 12, height = 10)

#Mean accessibility per cell type:

#Add the relative accessibility as metadata to the 
x.sp3.ext_peaks@metaData$norm.up_fl<- cell.rel_access_df$norm.up_fl
x.sp3.ext_peaks@metaData$norm.down_fl <- cell.rel_access_df$norm.down_fl
x.sp3.ext_peaks@metaData$enriched_up_fl_hsc <- cell.rel_access_df$up_enriched
x.sp3.ext_peaks@metaData$enriched_down_fl_hsc <- cell.rel_access_df$down_enriched
x.sp3.ext_peaks@metaData$NFE2_MkP_enriched <- cell.rel_access_df$NFE2_MkP_enriched
x.sp3.ext_peaks@metaData$norm.NFE2_MkP <- cell.rel_access_df$norm.NFE2_MkP
x.sp3.ext_peaks@metaData$NFE2_ery_enriched <- cell.rel_access_df$NFE2_ery_enriched
x.sp3.ext_peaks@metaData$gata1_mk_enriched <- cell.rel_access_df$gata1_mk_enriched
x.sp3.ext_peaks@metaData$norm.gata1_mk <- cell.rel_access_df$norm.gata1_mk

x.sp3.ext_peaks@metaData$gata1_ery_enriched<- cell.rel_access_df$gata1_ery_enriched
x.sp3.ext_peaks@metaData$iMK_unique_AREs_enriched<- cell.rel_access_df$iMK_unique_AREs_enriched
x.sp3.ext_peaks@metaData$ery_unique_AREs_enriched<- cell.rel_access_df$ery_unique_AREs_enriched


plotFeatureSingle( obj=x.sp3.ext_peaks,feature.value=x.sp3.ext_peaks@metaData$enriched_up_fl_hsc, method="umap", main="Relative accessibility over peaks up in bulk FL HSC",point.size=0.2, point.shape=19, quantiles=c(0.01, 0.99))

plotFeatureSingle( obj=x.sp3.ext_peaks,feature.value=x.sp3.ext_peaks@metaData$enriched_down_fl_hsc, method="umap", main="Relative accessibility over peaks down in bulk FL HSC",point.size=0.2, point.shape=19, quantiles=c(0.01, 0.99))

plotFeatureSingle( obj=x.sp3.ext_peaks,feature.value=x.sp3.ext_peaks@metaData$NFE2_MkP_enriched, method="umap", main="Relative accessibility over Nfe2 MkP unique peaks",point.size=0.2, point.shape=19, quantiles=c(0.01, 0.99))

plotFeatureSingle( obj=x.sp3.ext_peaks,feature.value=x.sp3.ext_peaks@metaData$NFE2_ery_enriched, method="umap", main="Relative accessibility over Nfe2 Ery unique peak",point.size=0.2, point.shape=19, quantiles=c(0.01, 0.99))

plotFeatureSingle( obj=x.sp3.ext_peaks,feature.value=x.sp3.ext_peaks@metaData$gata1_mk_enriched, method="umap", main="Relative accessibility over Gata1 unique MkP peaks",point.size=0.2, point.shape=19, quantiles=c(0.01, 0.99))

plotFeatureSingle( obj=x.sp3.ext_peaks,feature.value=x.sp3.ext_peaks@metaData$gata1_ery_enriched, method="umap", main="Relative accessibility over Gata1 unique Ery peaks",point.size=0.2, point.shape=19, quantiles=c(0.01, 0.99))

plotFeatureSingle( obj=x.sp3.ext_peaks,feature.value=x.sp3.ext_peaks@metaData$iMK_unique_AREs_enriched, method="umap", main="Relative accessibility over iMK unique AREs",point.size=0.2, point.shape=19, quantiles=c(0.01, 0.99))

plotFeatureSingle( obj=x.sp3.ext_peaks,feature.value=x.sp3.ext_peaks@metaData$ery_unique_AREs_enriched, method="umap", main="Relative accessibility over ery unique AREs",point.size=0.2, point.shape=19, quantiles=c(0.01, 0.99))


plotFeatureSingle( obj=x.sp3.ext_peaks,feature.value=x.sp3.ext_peaks@metaData$norm.down_fl, method="umap", main="",point.size=0.2, point.shape=19, quantiles=c(0.01, 0.99))
#Down_HSC
rmPmatFromSnap(x.sp3.ext_peaks)

setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/diff_peak_from_bulk_atac/Down_FL_HSC/")
peaks.names = system("ls | grep bed", intern=TRUE)
peak.gr.ls = lapply(peaks.names, function(x){
    peak.df = read.table(x)
    GRanges(peak.df[,1], IRanges(peak.df[,2], peak.df[,3]))
  })

peak.gr = reduce(Reduce(c, peak.gr.ls));
peak.gr
#Create a cell-by-peak matrix
library(GenomicRanges) #Step below is dependent on Genomic Ranges.

x.sp3.ext_peaks = createPmat(x.sp3.ext_peaks, peak.gr,  do.par = TRUE, num.cores = 12)

peak.matrix_down_bulk_fl <- as(object = x.sp3.ext_peaks@pmat, Class = 'matrix')


#   FL_HSC   CD41neg   CD41pos       MkP 
#-59.32616  86.74317  79.51511 -49.26705 

#NFE2 Mk unique
rmPmatFromSnap(x.sp3.ext_peaks)

setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/NFE2_Gata1_Meg_ChIP_analysis/NFE2_peaks_Mk_unique/")
peaks.names = system("ls | grep bed", intern=TRUE)
peak.gr.ls = lapply(peaks.names, function(x){
    peak.df = read.table(x)
    GRanges(peak.df[,1], IRanges(peak.df[,2], peak.df[,3]))
  })

peak.gr = reduce(Reduce(c, peak.gr.ls));
peak.gr
#Create a cell-by-peak matrix
library(GenomicRanges) #Step below is dependent on Genomic Ranges.

x.sp3.ext_peaks = createPmat(x.sp3.ext_peaks, peak.gr,  do.par = TRUE, num.cores = 8)

peak.matrix_NFE2_MkP <- as(object = x.sp3.ext_peaks@pmat, Class = 'matrix')



#NFE2 Ery unique
rmPmatFromSnap(x.sp3.ext_peaks)

setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/NFE2_Gata1_Meg_ChIP_analysis/NFE2_peaks_Ery_unique/")
peaks.names = system("ls | grep bed", intern=TRUE)
peak.gr.ls = lapply(peaks.names, function(x){
    peak.df = read.table(x)
    GRanges(peak.df[,1], IRanges(peak.df[,2], peak.df[,3]))
  })

peak.gr = reduce(Reduce(c, peak.gr.ls));
peak.gr
#Create a cell-by-peak matrix
library(GenomicRanges) #Step below is dependent on Genomic Ranges.

x.sp3.ext_peaks = createPmat(x.sp3.ext_peaks, peak.gr,  do.par = TRUE, num.cores = 8)

peak.matrix_NFE2_ery <- as(object = x.sp3.ext_peaks@pmat, Class = 'matrix')



#Gata1 Mk unique
rmPmatFromSnap(x.sp3.ext_peaks)

setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/NFE2_Gata1_Meg_ChIP_analysis/Gata1_peaks_Mk_unique/")
peaks.names = system("ls | grep bed", intern=TRUE)
peak.gr.ls = lapply(peaks.names, function(x){
    peak.df = read.table(x)
    GRanges(peak.df[,1], IRanges(peak.df[,2], peak.df[,3]))
  })

peak.gr = reduce(Reduce(c, peak.gr.ls));
peak.gr
#Create a cell-by-peak matrix
library(GenomicRanges) #Step below is dependent on Genomic Ranges.

x.sp3.ext_peaks = createPmat(x.sp3.ext_peaks, peak.gr,  do.par = TRUE, num.cores = 8)

peak.matrix_gata1_mk <- as(object = x.sp3.ext_peaks@pmat, Class = 'matrix')

#Cell based meanof accessibilities:

# FL_HSC   CD41neg   CD41pos       MkP 
#-22.60302 -16.49090 -11.73133  35.35945

#Gata1 ery unique
rmPmatFromSnap(x.sp3.ext_peaks)

setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/NFE2_Gata1_Meg_ChIP_analysis/Gata1_peaks_Ery_unique/")
peaks.names = system("ls | grep bed", intern=TRUE)
peak.gr.ls = lapply(peaks.names, function(x){
    peak.df = read.table(x)
    GRanges(peak.df[,1], IRanges(peak.df[,2], peak.df[,3]))
  })

peak.gr = reduce(Reduce(c, peak.gr.ls));
peak.gr
#Create a cell-by-peak matrix
library(GenomicRanges) #Step below is dependent on Genomic Ranges.

x.sp3.ext_peaks = createPmat(x.sp3.ext_peaks, peak.gr,  do.par = TRUE, num.cores = 8)

peak.matrix_gata1_ery <- as(object = x.sp3.ext_peaks@pmat, Class = 'matrix')


#iMK unique AREs
rmPmatFromSnap(x.sp3.ext_peaks)

setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/NFE2_Gata1_Meg_ChIP_analysis/bodine_paper_files/unique_iMK_AREs/")
peaks.names = system("ls | grep bed", intern=TRUE)
peak.gr.ls = lapply(peaks.names, function(x){
    peak.df = read.table(x)
    GRanges(peak.df[,1], IRanges(peak.df[,2], peak.df[,3]))
  })

peak.gr = reduce(Reduce(c, peak.gr.ls));
peak.gr
#Create a cell-by-peak matrix
library(GenomicRanges) #Step below is dependent on Genomic Ranges.

x.sp3.ext_peaks = createPmat(x.sp3.ext_peaks, peak.gr,  do.par = TRUE, num.cores = 8)

peak.matrix_iMK_unique_AREs <- as(object = x.sp3.ext_peaks@pmat, Class = 'matrix')

#Ery unique AREs
rmPmatFromSnap(x.sp3.ext_peaks)

setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/NFE2_Gata1_Meg_ChIP_analysis/bodine_paper_files/unique_ery_AREs/")
peaks.names = system("ls | grep bed", intern=TRUE)
peak.gr.ls = lapply(peaks.names, function(x){
    peak.df = read.table(x)
    GRanges(peak.df[,1], IRanges(peak.df[,2], peak.df[,3]))
  })

peak.gr = reduce(Reduce(c, peak.gr.ls));
peak.gr
#Create a cell-by-peak matrix
library(GenomicRanges) #Step below is dependent on Genomic Ranges.

x.sp3.ext_peaks = createPmat(x.sp3.ext_peaks, peak.gr,  do.par = TRUE, num.cores = 8)

peak.matrix_ery_unique_AREs <- as(object = x.sp3.ext_peaks@pmat, Class = 'matrix')
```

```{r visualization_of_external_peaks_overlapping_scatac_sample_peaks, echo=FALSE}

#Up FL_HSC vs FL_HSC_peaks.narrowPeak
rmPmatFromSnap(x.sp3.ext_peaks)

setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/peaks_sample_based_downscaled/FL_HSC_peaks.narrowPeak_FLHSC_up_bulk/")
peaks.names = system("ls | grep bed", intern=TRUE)
peak.gr.ls = lapply(peaks.names, function(x){
    peak.df = read.table(x)
    GRanges(peak.df[,1], IRanges(peak.df[,2], peak.df[,3]))
  })

peak.gr = reduce(Reduce(c, peak.gr.ls));
peak.gr
#Create a cell-by-peak matrix
library(GenomicRanges) #Step below is dependent on Genomic Ranges.

x.sp3.ext_peaks = createPmat(x.sp3.ext_peaks, peak.gr,  do.par = TRUE, num.cores = 8)
peak.matrix_up_bulk_fl_vs_fl <- as(object = x.sp3.ext_peaks@pmat, Class = 'matrix')

#Up FL_HSC vs CD41neg_peaks.narrowPeak
rmPmatFromSnap(x.sp3.ext_peaks)

setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/peaks_sample_based_downscaled/CD41neg_peaks.narrowPeak_FLHSC_up_bulk/")
peaks.names = system("ls | grep bed", intern=TRUE)
peak.gr.ls = lapply(peaks.names, function(x){
    peak.df = read.table(x)
    GRanges(peak.df[,1], IRanges(peak.df[,2], peak.df[,3]))
  })

peak.gr = reduce(Reduce(c, peak.gr.ls));
peak.gr
#Create a cell-by-peak matrix
library(GenomicRanges) #Step below is dependent on Genomic Ranges.

x.sp3.ext_peaks = createPmat(x.sp3.ext_peaks, peak.gr,  do.par = TRUE, num.cores = 8)
peak.matrix_up_bulk_fl_vs_CD41neg <- as(object = x.sp3.ext_peaks@pmat, Class = 'matrix')


#Up FL_HSC vs CD41pos_peaks.narrowPeak
rmPmatFromSnap(x.sp3.ext_peaks)

setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/peaks_sample_based_downscaled/CD41pos_peaks.narrowPeak_FLHSC_up_bulk/")
peaks.names = system("ls | grep bed", intern=TRUE)
peak.gr.ls = lapply(peaks.names, function(x){
    peak.df = read.table(x)
    GRanges(peak.df[,1], IRanges(peak.df[,2], peak.df[,3]))
  })

peak.gr = reduce(Reduce(c, peak.gr.ls));
peak.gr
#Create a cell-by-peak matrix
library(GenomicRanges) #Step below is dependent on Genomic Ranges.

x.sp3.ext_peaks = createPmat(x.sp3.ext_peaks, peak.gr,  do.par = TRUE, num.cores = 8)
peak.matrix_up_bulk_fl_vs_CD41pos <- as(object = x.sp3.ext_peaks@pmat, Class = 'matrix')

#Up FL_HSC vs MKP_peaks.narrowPeak
rmPmatFromSnap(x.sp3.ext_peaks)

setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/peaks_sample_based_downscaled/MkP_peaks.narrowPeak_FLHSC_up_bulk/")
peaks.names = system("ls | grep bed", intern=TRUE)
peak.gr.ls = lapply(peaks.names, function(x){
    peak.df = read.table(x)
    GRanges(peak.df[,1], IRanges(peak.df[,2], peak.df[,3]))
  })

peak.gr = reduce(Reduce(c, peak.gr.ls));
peak.gr
#Create a cell-by-peak matrix
library(GenomicRanges) #Step below is dependent on Genomic Ranges.

x.sp3.ext_peaks = createPmat(x.sp3.ext_peaks, peak.gr,  do.par = TRUE, num.cores = 8)
peak.matrix_up_bulk_fl_vs_MkP <- as(object = x.sp3.ext_peaks@pmat, Class = 'matrix')

#Process these files
peak.matrix_up_bulk_fl_vs_fl <- peak.matrix_up_bulk_fl_vs_fl[1:2403,]
peak.matrix_up_bulk_fl_vs_CD41neg <- peak.matrix_up_bulk_fl_vs_CD41neg[2404:4713,]
peak.matrix_up_bulk_fl_vs_CD41pos <- peak.matrix_up_bulk_fl_vs_CD41pos[4714:5836,]
peak.matrix_up_bulk_fl_vs_MkP <- peak.matrix_up_bulk_fl_vs_MkP[5837:8822,]

#TF-IFD normalize these

norm.peak.matrix_up_bulk_fl_vs_fl <- as.matrix(RunTFIDF(object =  t(peak.matrix_up_bulk_fl_vs_fl), method = 1) )
norm.peak.matrix_up_bulk_fl_vs_CD41neg <- as.matrix(RunTFIDF(object =  t(peak.matrix_up_bulk_fl_vs_CD41neg), method = 1) )
norm.peak.matrix_up_bulk_fl_vs_CD41pos <- as.matrix(RunTFIDF(object =  t(peak.matrix_up_bulk_fl_vs_CD41pos), method = 1) )
norm.peak.matrix_up_bulk_fl_vs_MkP <- as.matrix(RunTFIDF(object =  t(peak.matrix_up_bulk_fl_vs_MkP), method = 1) )

test1 <- as.data.frame( apply(norm.peak.matrix_up_bulk_fl_vs_fl,2,sum))
test2 <- as.data.frame( apply(norm.peak.matrix_up_bulk_fl_vs_CD41neg,2,sum))
test3 <- as.data.frame( apply(norm.peak.matrix_up_bulk_fl_vs_CD41pos,2,sum))
test4  <- as.data.frame( apply(norm.peak.matrix_up_bulk_fl_vs_MkP,2,sum))

colnames(test1)[1] <-"value"
colnames(test2)[1] <-"value"
colnames(test3)[1] <-"value"
colnames(test4)[1] <-"value"
df <- cbind(rbind(test1,test2,test3,test4),x.sp3.ext_peaks@sample)

colnames(df)<-c("value","sample")

plot_order_samples <- factor(c(rep(c('FL_HSC'), times=2403), rep(c('CD41neg'), times=2310), rep(c('CD41pos'), times=1123), rep(c('MkP'), times=2986)), levels=(c('FL_HSC', 'CD41neg', 'CD41pos', 'MkP')))
ggplot()+geom_violin(data = df, aes( y = value, x = plot_order_samples, fill = sample),draw_quantiles = c(0.25, 0.5, 0.75)) + theme_bw() + ggtitle("Gata1 MkP unique: summative accessibility") + xlab("Sample") + ylab("Summative accessibility")



```
Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
f

```{r variance_measure, echo=FALSE}
#Add bmat to snap but instead of making it binare I perform TF-IDF norm:
x.sp3.sample_peaks <- addBmatToSnap(x.sp3.sample_peaks)

#TF-IDF normalize bmat
#Cannot do this. Input to large. Have to do it over peaks
setwd("/mnt/data/common/jonas/scatac/191204_yuan_FLvsBM_HSC_BHHTF7DRXX/snap_atac_analysis/base_analysis_downscaled/peaks_sample_based_downscaled/")
peaks.names = system("ls | grep narrowPeak", intern=TRUE)
peak.gr.ls = lapply(peaks.names, function(x){
    peak.df = read.table(x)
    GRanges(peak.df[,1], IRanges(peak.df[,2], peak.df[,3]))
  })

peak.gr = reduce(Reduce(c, peak.gr.ls));
peak.gr


rmPmatFromSnap(x.sp3.sample_peaks)
x.sp3.sample_peaks = createPmat(x.sp3.sample_peaks, peak.gr,  do.par = TRUE, num.cores = 4)

norm.pmat <- t(TIFD_zscore_norm(as(object = x.sp3.sample_peaks@pmat, Class = 'matrix'), tfid_norm = 1))
#Calculate accessibility variance from the TF-IFD-norm peaks
norm.pmat_var <- as.data.frame( apply(norm.pmat,1,var, na.rm=T))
# Add in cell information
  norm.pmat_var[1:2403,2] <- 1
  norm.pmat_var[2404:4713,2] <-2
  norm.pmat_var[4714:5836,2] <- 3
  norm.pmat_var[5837:8822,2] <- 4
  
colnames(norm.pmat_var) <- c("var","cell_type")  

z1 <- apply(as.data.frame(norm.pmat_var[1:2403,1]),2,scale)
z2 <-apply(as.data.frame(norm.pmat_var[2404:4713,1]),2,scale)
z3 <-apply(as.data.frame(norm.pmat_var[4714:5836,1]),2,scale)
z4 <- apply(as.data.frame(norm.pmat_var[5837:8822,1]),2,scale)


norm.pmat_var[1:2403,3] <- z1
norm.pmat_var[2404:4713,3] <-z2
norm.pmat_var[4714:5836,3] <- z3
norm.pmat_var[5837:8822,3] <- z4
  
colnames(norm.pmat_var)[3] <- "z_var"

library(ggplot2)
ggplot(norm.pmat_var,aes(var, colour=factor(cell_type)))+stat_ecdf()+theme_bw()+theme(legend.position = c(0.8,0.25),legend.title=element_blank())+theme(legend.position = "bottom",legend.title=element_blank(), legend.key=element_rect(colour="NA", fill = "NA")) 


ggplot(z_var,aes(grp.z, colour=factor(cell_type)))+stat_ecdf()+theme_bw()+theme(legend.position = c(0.8,0.25),legend.title=element_blank())+theme(legend.position = "bottom",legend.title=element_blank(), legend.key=element_rect(colour="NA", fill = "NA")) 

#+scale_color_manual(values=c("0"="black","1"="blue", "10"="red", "11"="orange"))


```


